--- nagios-3.5.0/cgi/cgiutils.c.orig	2013-03-09 13:46:35.000000000 -0800
+++ nagios-3.5.0/cgi/cgiutils.c	2013-03-18 18:00:33.000000000 -0700
@@ -1568,7 +1568,6 @@
 	char cgi_ssi_file[MAX_INPUT_BUFFER];
 	char raw_cgi_name[MAX_INPUT_BUFFER];
 	char *stripped_cgi_name;
-	int x;
 
 	/* common header or footer */
 	snprintf(common_ssi_file, sizeof(common_ssi_file) - 1, "%scommon-%s.ssi", physical_ssi_path, (type == SSI_HEADER) ? "header" : "footer");
