--- nagios-3.5.0/cgi/extinfo.c.orig	2011-08-17 00:36:27.000000000 -0700
+++ nagios-3.5.0/cgi/extinfo.c	2013-03-18 14:06:15.000000000 -0700
@@ -130,6 +130,9 @@
 int embedded = FALSE;
 int display_header = TRUE;
 
+/* Establish that this patch is in place. */
+char refresh_patch_ident[] = "$RefreshPatchCompileTime: " __TIME__ " on " __DATE__ " (" __FILE__ ") $";
+
 
 
 int main(void) {
@@ -587,7 +590,7 @@
 
 	printf("Cache-Control: no-store\r\n");
 	printf("Pragma: no-cache\r\n");
-	printf("Refresh: %d\r\n", refresh_rate);
+	// printf("Refresh: %d\r\n", refresh_rate);
 
 	time(&current_time);
 	get_time_string(&current_time, date_time, (int)sizeof(date_time), HTTP_DATE_TIME);
@@ -615,7 +618,8 @@
 		}
 	printf("</head>\n");
 
-	printf("<body CLASS='extinfo'>\n");
+	printf("<body CLASS='extinfo'"
+	       " onload='setTimeout(\"location=self.location;\",%d)'>\n", (refresh_rate ? refresh_rate : 1000000) * 1000);
 
 	/* include user SSI header */
 	include_ssi_files(EXTINFO_CGI, SSI_HEADER);
--- nagios-3.5.0/cgi/outages.c.orig	2011-08-17 00:36:27.000000000 -0700
+++ nagios-3.5.0/cgi/outages.c	2013-03-18 14:04:42.000000000 -0700
@@ -205,7 +205,7 @@
 
 	printf("Cache-Control: no-store\r\n");
 	printf("Pragma: no-cache\r\n");
-	printf("Refresh: %d\r\n", refresh_rate);
+	// printf("Refresh: %d\r\n", refresh_rate);
 
 	time(&current_time);
 	get_time_string(&current_time, date_time, (int)sizeof(date_time), HTTP_DATE_TIME);
@@ -234,7 +234,8 @@
 
 	printf("</head>\n");
 
-	printf("<body CLASS='outages'>\n");
+	printf("<body CLASS='outages'"
+	       " onload='setTimeout(\"location=self.location;\",%d)'>\n", (refresh_rate ? refresh_rate : 1000000) * 1000);
 
 	/* include user SSI header */
 	include_ssi_files(OUTAGES_CGI, SSI_HEADER);
--- nagios-3.5.0/cgi/status.c.orig	2013-03-09 13:46:35.000000000 -0800
+++ nagios-3.5.0/cgi/status.c	2013-03-18 14:04:42.000000000 -0700
@@ -529,7 +529,7 @@
 
 	printf("Cache-Control: no-store\r\n");
 	printf("Pragma: no-cache\r\n");
-	printf("Refresh: %d\r\n", refresh_rate);
+	// printf("Refresh: %d\r\n", refresh_rate);
 
 	get_time_string(&current_time, date_time, (int)sizeof(date_time), HTTP_DATE_TIME);
 	printf("Last-Modified: %s\r\n", date_time);
@@ -565,7 +565,8 @@
 
 	printf("</head>\n");
 
-	printf("<body class='status'>\n");
+	printf("<body class='status'"
+	       " onload='setTimeout(\"location=self.location;\",%d)'>\n", (refresh_rate ? refresh_rate : 1000000) * 1000);
 
 	/* include user SSI header */
 	include_ssi_files(STATUS_CGI, SSI_HEADER);
--- nagios-3.5.0/cgi/statusmap.c.orig	2011-08-17 00:36:27.000000000 -0700
+++ nagios-3.5.0/cgi/statusmap.c	2013-03-18 14:04:42.000000000 -0700
@@ -318,7 +318,7 @@
 	if(create_type == CREATE_HTML) {
 		printf("Cache-Control: no-store\r\n");
 		printf("Pragma: no-cache\r\n");
-		printf("Refresh: %d\r\n", refresh_rate);
+		// printf("Refresh: %d\r\n", refresh_rate);
 
 		time(&current_time);
 		get_time_string(&current_time, date_time, sizeof(date_time), HTTP_DATE_TIME);
@@ -350,7 +350,8 @@
 
 		printf("</head>\n");
 
-		printf("<body CLASS='statusmap' name='mappage' id='mappage'>\n");
+		printf("<body CLASS='statusmap' name='mappage' id='mappage'"
+		       " onload='setTimeout(\"location=self.location;\",%d)'>\n", (refresh_rate ? refresh_rate : 1000000) * 1000);
 
 		/* include user SSI header */
 		include_ssi_files(STATUSMAP_CGI, SSI_HEADER);
--- nagios-3.5.0/cgi/tac.c.orig	2012-11-12 10:33:37.000000000 -0800
+++ nagios-3.5.0/cgi/tac.c	2013-03-18 14:04:42.000000000 -0700
@@ -371,7 +371,7 @@
 
 	printf("Cache-Control: no-store\r\n");
 	printf("Pragma: no-cache\r\n");
-	printf("Refresh: %d\r\n", refresh_rate);
+	// printf("Refresh: %d\r\n", refresh_rate);
 
 	time(&current_time);
 	get_time_string(&current_time, date_time, (int)sizeof(date_time), HTTP_DATE_TIME);
@@ -399,7 +399,8 @@
 		}
 
 	printf("</HEAD>\n");
-	printf("<BODY CLASS='tac' marginwidth=2 marginheight=2 topmargin=0 leftmargin=0 rightmargin=0>\n");
+	printf("<BODY CLASS='tac' marginwidth=2 marginheight=2 topmargin=0 leftmargin=0 rightmargin=0"
+	       " onload='setTimeout(\"location=self.location;\",%d)'>\n", (refresh_rate ? refresh_rate : 1000000) * 1000);
 
 	/* include user SSI header */
 	include_ssi_files(TAC_CGI, SSI_HEADER);
