--- nagios-3.5.0/cgi/avail.c.orig	2013-08-30 10:46:14.000000000 -0700
+++ nagios-3.5.0/cgi/avail.c	2016-03-06 22:05:50.000000000 -0800
@@ -2260,7 +2260,7 @@
 	time_t part_start_time = 0L;
 	time_t part_subject_state = 0L;
 	int saved_status = 0;
-	int saved_stamp = 0;
+	time_t saved_stamp = 0;
 	int count = 0;
 	archived_state *temp_before = NULL;
 	archived_state *last = NULL;
@@ -2364,6 +2364,10 @@
 				}
 			saved_status = temp_as->entry_type;
 			saved_stamp = temp_as->time_stamp;
+
+			/* check if first time is before schedule downtime */
+			if(saved_stamp < start_time)
+				saved_stamp = start_time;
 			}
 		}
 
