--- nagios-3.5.0/common/macros.c.orig	2013-03-09 13:46:35.000000000 -0800
+++ nagios-3.5.0/common/macros.c	2013-03-18 17:53:39.000000000 -0700
@@ -22,8 +22,8 @@
  *
  *****************************************************************************/
 
-#include "../include/macros.h"
 #include "../include/config.h"
+#include "../include/macros.h"
 #include "../include/common.h"
 #include "../include/objects.h"
 #include "../include/statusdata.h"
