diff -Naur nagios-2.10/include/objects.h nagios-2.10-patched/include/objects.h
--- nagios-2.5/include/objects.h	2006-03-21 14:21:51.000000000 -0800
+++ nagios-2.10-patched/include/objects.h	2007-04-18 09:05:19.000000000 -0700
@@ -45,7 +45,7 @@
 
 #define MAX_HOSTNAME_LENGTH            		64	/* max. host name length */
 #define MAX_SERVICEDESC_LENGTH			64	/* max. service description length */
-#define MAX_PLUGINOUTPUT_LENGTH			332	/* max. length of plugin output */
+#define MAX_PLUGINOUTPUT_LENGTH			2048	/* max. length of plugin output : changed via patch from tdondich@groundworkopensource.com */
 
 #define MAX_STATE_HISTORY_ENTRIES		21	/* max number of old states to keep track of for flap detection */
 
