*** noma_2.0.3/contrib/configconvert.php.orig	Mon Nov 21 03:25:24 2011
--- noma_2.0.3/contrib/configconvert.php	Mon Aug 20 14:18:22 2012
***************
*** 58,63 ****
--- 58,64 ----
  
  $conf['db']['sqlite3']['type'] = 'sqlite3';
  $conf['db']['mysql']['type'] = 'mysql';
+ $conf['db']['postgresql']['type'] = 'postgresql';
  $conf['frontend']['language'] =     $language;
  
  
***************
*** 77,83 ****
      $conf['frontend']['debug']['queries'] = FALSE;
  }
  $conf['api'] =     $dbNDO;
! $conf['api']['type'] = 'mysql';
      
      
  $conf['frontend']['authentication_type'] =     $authentication_type;
--- 78,84 ----
      $conf['frontend']['debug']['queries'] = FALSE;
  }
  $conf['api'] =     $dbNDO;
! $conf['api']['type'] = 'postgresql';
      
      
  $conf['frontend']['authentication_type'] =     $authentication_type;
