*** noma_2.0.3/sql/mysql/mysql_new_install.sql.orig	Thu Jan  8 08:13:55 2015
--- noma_2.0.3/sql/mysql/mysql_new_install.sql	Thu Jan 29 14:25:28 2015
***************
*** 126,132 ****
  
  CREATE TABLE IF NOT EXISTS `notifications` (
    `id` int(11) NOT NULL AUTO_INCREMENT,
!   `notification_name` varchar(40) NULL,
    `notification_description` varchar(1024) NULL,
    `active` tinyint(1) NOT NULL,
    `username` varchar(255) NOT NULL,
--- 126,132 ----
  
  CREATE TABLE IF NOT EXISTS `notifications` (
    `id` int(11) NOT NULL AUTO_INCREMENT,
!   `notification_name` varchar(255) NULL,
    `notification_description` varchar(1024) NULL,
    `active` tinyint(1) NOT NULL,
    `username` varchar(255) NOT NULL,
