*** noma_2.0.3/share/inc/content_funcs.php.orig	Mon Nov 21 03:25:24 2011
--- noma_2.0.3/share/inc/content_funcs.php	Thu Sep 20 00:19:11 2012
***************
*** 378,389 ****
  			'auth_type_required'	=> false,
  			'title'					=> NAVIGATION_LOGS,
  		),
! 		array(
! 			'actions'				=> array('logout'),
! 			'admin_only'			=> false,
! 			'auth_type_required'	=> true,
! 			'title'					=> NAVIGATION_LOGOUT,
! 		),
  	);
  
  	$separator = '&nbsp;&nbsp;&nbsp;';
--- 378,389 ----
  			'auth_type_required'	=> false,
  			'title'					=> NAVIGATION_LOGS,
  		),
! #		array(
! #			'actions'				=> array('logout'),
! #			'admin_only'			=> false,
! #			'auth_type_required'	=> true,
! #			'title'					=> NAVIGATION_LOGOUT,
! #		),
  	);
  
  	$separator = '&nbsp;&nbsp;&nbsp;';
