*** noma_2.0.3/configure.ac.orig	Wed May 30 16:44:08 2012
--- noma_2.0.3/configure.ac	Mon Aug 20 12:31:45 2012
***************
*** 62,74 ****
  AX_PROG_PERL_MODULES([YAML::Syck Getopt::Long Pod::Usage Digest::MD5 POSIX Data::Dumper Thread::Queue IO::Select IO::Socket Fcntl FindBin DBI DateTime::TimeZone DateTime::Format::Strptime Date::Calc Hash::Merge::Simple], , AC_MSG_ERROR([Missing perl modules]))
  AX_PROG_PERL_MODULES([DBD::SQLite], HAVESQLITE="yes", HAVESQLITE="no")
  AX_PROG_PERL_MODULES([DBD::mysql], HAVEMYSQL="yes", HAVEMYSQL="no")
  
! if test "$HAVESQLITE" = "no" && test "$HAVEMYSQL" = "no"; then
!     AC_MSG_ERROR([You need either DBD::SQLite or DBD::mysql])
  fi
  
  if test "$HAVESQLITE" = "no"; then
!     AC_MSG_WARN([DBD::SQLite is not installed: if you want to use MySQL you will need to configure your NoMa by hand])
  fi
  
  
--- 62,75 ----
  AX_PROG_PERL_MODULES([YAML::Syck Getopt::Long Pod::Usage Digest::MD5 POSIX Data::Dumper Thread::Queue IO::Select IO::Socket Fcntl FindBin DBI DateTime::TimeZone DateTime::Format::Strptime Date::Calc Hash::Merge::Simple], , AC_MSG_ERROR([Missing perl modules]))
  AX_PROG_PERL_MODULES([DBD::SQLite], HAVESQLITE="yes", HAVESQLITE="no")
  AX_PROG_PERL_MODULES([DBD::mysql], HAVEMYSQL="yes", HAVEMYSQL="no")
+ AX_PROG_PERL_MODULES([DBD::Pg], HAVEPOSTGRESQL="yes", HAVEPOSTGRESQL="no")
  
! if test "$HAVESQLITE" = "no" && test "$HAVEMYSQL" = "no" && test "$HAVEPOSTGRESQL" = "no"; then
!     AC_MSG_ERROR([You need either DBD::SQLite or DBD::mysql or DBD::Pg])
  fi
  
  if test "$HAVESQLITE" = "no"; then
!     AC_MSG_WARN([DBD::SQLite is not installed: if you want to use MySQL or PostgreSQL you will need to configure your NoMa by hand])
  fi
  
  
