*** noma_2.0.3/sql/sqlite3/install/default_data.sql.orig	Mon Nov 21 03:25:25 2011
--- noma_2.0.3/sql/sqlite3/install/default_data.sql	Thu Sep 20 00:25:24 2012
***************
*** 1,9 ****
  INSERT INTO `contactgroups` (`id`, `name_short`, `name`, `view_only`, `timeframe_id`, `timezone_id`) VALUES(1, 'group1', 'Group 1', 0, 1, 305);
  INSERT INTO `contactgroups_to_contacts` (`contactgroup_id`, `contact_id`) VALUES(1, 2);
  INSERT INTO `contacts` (`id`, `admin`, `username`, `full_name`, `email`, `phone`, `mobile`, `section`, `growladdress`, `password`, `timeframe_id`, `timezone_id`, `restrict_alerts`) VALUES(1, 0, '[---]', '', '', '', '', '', '', '', 0, 0, NULL);
! INSERT INTO `contacts` (`id`, `admin`, `username`, `full_name`, `email`, `phone`, `mobile`, `section`, `growladdress`, `password`, `timeframe_id`, `timezone_id`, `restrict_alerts`) VALUES(2, 1, 'nagiosadmin', 'Nagios Administrator', 'nagios@localhost', '', '', '', '192.168.1.109', '9e2b1592bd13bea759dab1e3011cab7ef47930cd', 1, 0, 0);
  INSERT INTO `information` (`id`, `type`, `content`) VALUES(0, 'dbversion', '2000');
! INSERT INTO `notifications` (`id`, `notification_name`, `notification_description`, `active`, `username`, `recipients_include`, `recipients_exclude`, `hosts_include`, `hosts_exclude`, `hostgroups_include`, `hostgroups_exclude`, `services_include`, `services_exclude`, `servicegroups_include`, `servicegroups_exclude`, `notify_after_tries`, `let_notifier_handle`, `rollover`, `reloop_delay`, `on_ok`, `on_warning`, `on_unknown`, `on_host_unreachable`, `on_critical`, `on_host_up`, `on_host_down`, `on_type_problem`, `on_type_recovery`, `on_type_flappingstart`, `on_type_flappingstop`, `on_type_flappingdisabled`, `on_type_downtimestart`, `on_type_downtimeend`, `on_type_downtimecancelled`, `on_type_acknowledgement`, `on_type_custom`, `timezone_id`, `timeframe_id`) VALUES(1, 'default', 'default rule', 1, 'nagiosadmin', '', '', '*', '', '*', '', '*', '', '*', '', '1', 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1);
  INSERT INTO `notifications_to_contacts` (`notification_id`, `contact_id`) VALUES(1, 2);
  INSERT INTO `notifications_to_methods` (`notification_id`, `method_id`) VALUES(1, 1);
  INSERT INTO `notification_methods` (`id`, `method`, `command`, `contact_field`, `sender`, `on_fail`, `ack_able`) VALUES(5, 'Voice + SMS fallback', 'voicecall', 'phone', '', 2, 1);
--- 1,9 ----
  INSERT INTO `contactgroups` (`id`, `name_short`, `name`, `view_only`, `timeframe_id`, `timezone_id`) VALUES(1, 'group1', 'Group 1', 0, 1, 305);
  INSERT INTO `contactgroups_to_contacts` (`contactgroup_id`, `contact_id`) VALUES(1, 2);
  INSERT INTO `contacts` (`id`, `admin`, `username`, `full_name`, `email`, `phone`, `mobile`, `section`, `growladdress`, `password`, `timeframe_id`, `timezone_id`, `restrict_alerts`) VALUES(1, 0, '[---]', '', '', '', '', '', '', '', 0, 0, NULL);
! INSERT INTO `contacts` (`id`, `admin`, `username`, `full_name`, `email`, `phone`, `mobile`, `section`, `growladdress`, `password`, `timeframe_id`, `timezone_id`, `restrict_alerts`) VALUES(2, 1, 'admin', 'Nagios Administrator', 'nagios@localhost', '', '', '', '192.168.1.109', '9e2b1592bd13bea759dab1e3011cab7ef47930cd', 1, 0, 0);
  INSERT INTO `information` (`id`, `type`, `content`) VALUES(0, 'dbversion', '2000');
! INSERT INTO `notifications` (`id`, `notification_name`, `notification_description`, `active`, `username`, `recipients_include`, `recipients_exclude`, `hosts_include`, `hosts_exclude`, `hostgroups_include`, `hostgroups_exclude`, `services_include`, `services_exclude`, `servicegroups_include`, `servicegroups_exclude`, `notify_after_tries`, `let_notifier_handle`, `rollover`, `reloop_delay`, `on_ok`, `on_warning`, `on_unknown`, `on_host_unreachable`, `on_critical`, `on_host_up`, `on_host_down`, `on_type_problem`, `on_type_recovery`, `on_type_flappingstart`, `on_type_flappingstop`, `on_type_flappingdisabled`, `on_type_downtimestart`, `on_type_downtimeend`, `on_type_downtimecancelled`, `on_type_acknowledgement`, `on_type_custom`, `timezone_id`, `timeframe_id`) VALUES(1, 'default', 'default rule', 1, 'admin', '', '', '*', '', '*', '', '*', '', '*', '', '1', 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1);
  INSERT INTO `notifications_to_contacts` (`notification_id`, `contact_id`) VALUES(1, 2);
  INSERT INTO `notifications_to_methods` (`notification_id`, `method_id`) VALUES(1, 1);
  INSERT INTO `notification_methods` (`id`, `method`, `command`, `contact_field`, `sender`, `on_fail`, `ack_able`) VALUES(5, 'Voice + SMS fallback', 'voicecall', 'phone', '', 2, 1);
