--- nedi/cusdi.pl.orig	2016-10-26 20:16:55.000000000 -0700
+++ nedi/cusdi.pl	2016-10-27 08:06:56.000000000 -0700
@@ -188,7 +188,7 @@
 							my $dhc = $session->var_bind_list()->{$oid[2]};
 							my $vid = $session->var_bind_list()->{$oid[3]} if defined $oid[3];
 							$ext =~ s/"//g;
-							$vid = 0 unless defined $vid an $vid =~ /^\d+$/;
+							$vid = 0 unless defined $vid and $vid =~ /^\d+$/;
 							$des .= " $dhc" unless $dhc =~ /noSuchObject/;
 
 							$out .= "$doip{$id}\t$nam $community";
