--- nedi/html/inc/header.php.orig	2016-04-09 04:31:25.000000000 -0700
+++ nedi/html/inc/header.php	2016-09-19 14:12:19.193218006 -0700
@@ -267,6 +267,7 @@
 	if($debug){
 		ini_set('display_errors', 'On');
 		error_reporting(E_ALL);
+		echo "<h1>Debug Info</h1>\n";
 		echo "<div class=\"textpad code pre good half\">Self:		$self\n";
 		echo "Version:	$_SESSION[ver]\n";
 		echo "NeDipath:	$nedipath\n";
@@ -285,7 +286,8 @@
 }
 
 if( strpos($_SESSION['group'],$modgroup[$self]) === false){
-	echo "<br><br><br><br><div class=\"textpad txta half\">$nokmsg</div><br>\n";
+	echo "<h1>$acslbl ".$mlvl['100']."</h1>\n";
+	echo "<div class=\"textpad txta half\">$nokmsg</div>\n";
 	include_once ("inc/footer.php");
 	die;
 }
