--- nedi/html/System-Database.php.orig	2016-04-09 04:31:25.000000000 -0700
+++ nedi/html/System-Database.php	2016-09-19 14:27:45.337141188 -0700
@@ -148,7 +148,7 @@
 	?>
 	</select>
 </td>
-<td class="ctr s">
+<td class="ctr top s">
 	<h3><?= $dstlbl ?></h3>
 	<select size="1" name="type">
 		<option value="htm" <?= ($type=="htm")?" selected":"" ?>>html</option>
@@ -238,7 +238,7 @@
 		echo "<table class=\"content\">\n\t<tr class=\"bgsub\">\n";
 		for ($i = 0; $i < DbNumFields($res); ++$i) {
 			$field = DbFieldName($res, $i);
-			echo  "\t\t<th>$i $field</th>\n";
+			echo  "\t\t<th class=\"left\">$i $field</th>\n";
 		}
 		echo "\t</tr>\n";
 		$row = 0;
