--- nedi/html/Devices-Config.php.orig	2016-04-09 04:31:25.000000000 -0700
+++ nedi/html/Devices-Config.php	2016-09-19 14:45:22.849273425 -0700
@@ -62,7 +62,7 @@
 	<a href="<?= $self ?>.php"><img src="img/32/<?= $selfi ?>.png" title="<?= $self ?>"></a>
 </td>
 <td class="ctr b">
-	<?= $lstlbl ?><p>
+	<h3><?= $lstlbl ?></h3><p>
 	<select size="1" name="shl">
 		<option value="I"><?= substr($cfglbl,0,6) ?> ~
 		<option value="E" <?= ($shl == "E")?" selected":"" ?>><?= substr($cfglbl,0,6) ?> !~
@@ -81,7 +81,7 @@
 	</select>
 </td>
 <td class="ctr b">
-	<?= $cmplbl ?><p>
+	<h3><?= $cmplbl ?></h3><p>
 	<select size="1" name="dd">
 		<option value="">- <?= $typlbl ?> -
 <?php
@@ -102,7 +102,7 @@
 	</select>
 </td>
 <td class="ctr b">
-	<?= $limlbl ?><p>
+	<h3><?= $limlbl ?></h3><p>
 	<select size="1" name="lim" title="<?= $cfglbl ?>/<?= $chglbl ?>">
 <?php selectbox("limit",$lic) ?>
 	</select>
