--- nedi/html/Devices-Write.php.orig	2016-04-09 04:31:25.000000000 -0700
+++ nedi/html/Devices-Write.php	2016-09-19 14:58:36.377841361 -0700
@@ -67,12 +67,12 @@
 </tr><tr class="bgsub">
 <td></td>
 <td class="top ctr b">
-	<?= $cmdlbl ?> / <?= $cfglbl ?>
+	<h3><?= $cmdlbl ?> / <?= $cfglbl ?></h3>
 	<p>
 	<textarea rows="6" name="cmd" cols="60"><?= $cmd ?></textarea>
 </td>
 <td class="top ctr b">
-	Interface <?= $cfglbl ?>
+	<h3>Interface <?= $cfglbl ?></h3>
 	<p>
 	<select size="1" name="int">
 		<option value=""><?= $sellbl ?> ->
