--- nedi/html/Nodes-Status.php.orig	2016-04-09 04:31:25.000000000 -0700
+++ nedi/html/Nodes-Status.php	2016-09-19 15:08:23.274630339 -0700
@@ -32,7 +32,7 @@
 	<a href="<?= $self ?>.php"><img src="img/32/<?= $selfi ?>.png" title="<?= $self ?>"></a>
 </td>
 <td class="ctr">
-	<h3>
+	<!-- <h3> -->
 		<select name="in">
 			<option value="mac">MAC <?= $adrlbl ?>
 
@@ -42,7 +42,7 @@
 
 		</select>
 		<input type="text" name="st" value="<?= $st ?>" class="m">
-	</h3>
+	<!-- </h3> -->
 </td>
 <td class="ctr">
 <?php if($rrdcmd and $_SESSION['gsiz']){ ?>
