--- nagios-4.2.4/cgi/Makefile.in.orig	2016-12-07 08:31:10.000000000 -0800
+++ nagios-4.2.4/cgi/Makefile.in	2017-01-21 03:49:25.017124803 -0800
@@ -40,7 +40,7 @@
 LDFLAGS=@LDFLAGS@
 LIBS=@LIBS@
 
-CGIS=archivejson.cgi avail.cgi cmd.cgi config.cgi extinfo.cgi history.cgi notifications.cgi objectjson.cgi outages.cgi showlog.cgi status.cgi statusjson.cgi statuswml.cgi summary.cgi tac.cgi $(CGIEXTRAS)
+CGIS=archivejson.cgi avail.cgi cmd.cgi config.cgi extinfo.cgi history.cgi notifications.cgi objectjson.cgi outages.cgi showlog.cgi status.cgi statusjson.cgi statuswml.cgi statuswml2.cgi summary.cgi tac.cgi $(CGIEXTRAS)
 
 # Object functions
 ODATALIBS=objects-cgi.o xobjects-cgi.o
@@ -144,6 +144,9 @@
 statuswml.cgi: statuswml.c $(CGIDEPS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ statuswml.c $(CGILIBS) $(LIBS)
 
+statuswml2.cgi: statuswml2.c $(CGIDEPS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ statuswml2.c $(CGILIBS) $(LIBS)
+
 statusmap.cgi: statusmap.c $(CGIDEPS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ statusmap.c $(CGILIBS) $(GDLIBS) $(LIBS)
 
