*** weathermap.orig/random-bits/auto-overlib.pl	Mon Mar 12 16:07:46 2012
--- weathermap/random-bits/auto-overlib.pl	Mon Apr 30 17:40:01 2012
***************
*** 1,16 ****
! #!/usr/bin/perl
  
  use DBI;
  
  # This should be the URL for the base of your cacti install (no trailing slash)
! $cacti_base = "http://www.mynet.net/cacti";
  
  # How we should access your Cacti database....
  $dbname = "cacti";
  $dbuser = "cactiuser";
! $dbpass = "somepassword";
  $dbhost = "localhost";
! $dbtype = "mysql";  # "mysql" or "postgresql"
  
  #
  # You shouldn't need to change anything below here
--- 1,17 ----
! #!/usr/local/groundwork/perl/bin/perl
  
  use DBI;
  
  # This should be the URL for the base of your cacti install (no trailing slash)
! # $cacti_base = "http://www.mynet.net/cacti";
! $cacti_base = "http://myhost.mydomain.com/portal/auth/portal/groundwork-monitor/nagios/Cacti";
  
  # How we should access your Cacti database....
  $dbname = "cacti";
  $dbuser = "cactiuser";
! $dbpass = "cactiuser";
  $dbhost = "localhost";
! $dbtype = "postgresql";  # "mysql" or "postgresql"
  
  #
  # You shouldn't need to change anything below here
