--- nedi/inc/libdb.pm.orig	2017-05-16 02:34:51.000000000 -0700
+++ nedi/inc/libdb.pm	2017-08-28 13:48:48.000000000 -0700
@@ -157,7 +157,7 @@
 				$index)" );
 
 			print "cables\n";
-			$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(panel), INDEX(jack), INDEX(nbrpanel), INDEX(nbrjack), INDEX(cblopt), PRIMARY KEY(id)';
+			$index = ($misc::backend eq 'Pg')?'':', INDEX(panel), INDEX(jack), INDEX(nbrpanel), INDEX(nbrjack), INDEX(cblopt), PRIMARY KEY(id)';
 			$dbh->do("CREATE TABLE cables(
 				id $serid,
 				panel $vchar(64) NOT NULL,
@@ -171,7 +171,7 @@
 				$index)" );
 
 			print "translations\n";
-			$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(srctype)';
+			$index = ($misc::backend eq 'Pg')?'':', INDEX(srctype), PRIMARY KEY(id)';
 			$dbh->do("CREATE TABLE translations(
 				id $serid,
 				srctype $vchar(32)  DEFAULT '',
@@ -186,28 +186,104 @@
 
 			print "\nModifying tables\n";
 			if($misc::backend eq 'Pg'){
-				print "\nIndexing new tables\n";
-				$dbh->do('CREATE INDEX vlans_device	ON vlanport	USING btree (device);');
-				$dbh->do('CREATE INDEX vlans_vlanid	ON vlanport	USING btree (vlanid);');
-				$dbh->do('CREATE INDEX vlans_ifname	ON vlanport	USING btree (vlanif);');
-				$dbh->do('CREATE INDEX vlans_vlopts	ON vlanport	USING btree (vlopts);');
-				$dbh->do('CREATE INDEX cables_panel	ON cables	USING btree (panel);');
-				$dbh->do('CREATE INDEX cables_jack	ON cables	USING btree (jack);');
-				$dbh->do('CREATE INDEX cables_nbrpanel	ON cables	USING btree (nbrpanel);');
-				$dbh->do('CREATE INDEX cables_nbrjack	ON cables	USING btree (nbrjack);');
-				$dbh->do('CREATE INDEX cables_cblopt	ON cables	USING btree (patopt);');
-				$dbh->do('CREATE INDEX translations_id	ON translations USING btree (id);');
-				$dbh->do('CREATE INDEX translations_sty	ON translations USING btree (srctype);');
-				$dbh->do("ALTER TABLE policies ALTER type TYPE $vchar(128);");
+				$dbh->do('ALTER TABLE ONLY cables       ADD CONSTRAINT cables_pkey       PRIMARY KEY (id);');
+				$dbh->do('ALTER TABLE ONLY configs      ADD CONSTRAINT configs_pkey      PRIMARY KEY (device);');
+				$dbh->do('ALTER TABLE ONLY devices      ADD CONSTRAINT devices_pkey      PRIMARY KEY (device);');
+				$dbh->do('ALTER TABLE ONLY inventory    ADD CONSTRAINT inventory_pkey    PRIMARY KEY (serial);');
+				$dbh->do('ALTER TABLE ONLY monimap      ADD CONSTRAINT monimap_pkey      PRIMARY KEY (id);');
+				$dbh->do('ALTER TABLE ONLY monitoring   ADD CONSTRAINT monitoring_pkey   PRIMARY KEY (name);');
+				$dbh->do('ALTER TABLE ONLY system       ADD CONSTRAINT system_pkey       PRIMARY KEY (name);');
+				$dbh->do('ALTER TABLE ONLY translations ADD CONSTRAINT translations_pkey PRIMARY KEY (id);');
+				$dbh->do('ALTER TABLE ONLY users        ADD CONSTRAINT users_pkey        PRIMARY KEY (usrname);');
+
+				$dbh->do('DROP INDEX IF EXISTS configs_device;');
+				$dbh->do('DROP INDEX IF EXISTS devices_device;');
+				$dbh->do('DROP INDEX IF EXISTS dns6_dns6update;');
+				$dbh->do('DROP INDEX IF EXISTS dns_dnsupdate;');
+				$dbh->do('DROP INDEX IF EXISTS install_name;');
+				$dbh->do('DROP INDEX IF EXISTS interfaces_device;');
+				$dbh->do('DROP INDEX IF EXISTS interfaces_devif;');
+				$dbh->do('DROP INDEX IF EXISTS inventory_serial;');
+				$dbh->do('DROP INDEX IF EXISTS iptrack_dev;');
+				$dbh->do('DROP INDEX IF EXISTS iptrack_if;');
+				$dbh->do('DROP INDEX IF EXISTS monimap_usr;');
+				$dbh->do('DROP INDEX IF EXISTS monitoring_name;');
+				$dbh->do('DROP INDEX IF EXISTS nodarp_ipupdate;');
+				$dbh->do('DROP INDEX IF EXISTS nodarp_srvupdate;');
+				$dbh->do('DROP INDEX IF EXISTS nodnd_ip6update;');
+				$dbh->do('DROP INDEX IF EXISTS nodnd_srv6update;');
+				$dbh->do('DROP INDEX IF EXISTS system_name;');
+
+				$dbh->do('ALTER TABLE ONLY configs    DROP CONSTRAINT IF EXISTS configs_device_key;');
+				$dbh->do('ALTER TABLE ONLY devices    DROP CONSTRAINT IF EXISTS devices_device_key;');
+				$dbh->do('ALTER TABLE ONLY install    DROP CONSTRAINT IF EXISTS install_name_key;');
+				$dbh->do('ALTER TABLE ONLY inventory  DROP CONSTRAINT IF EXISTS inventory_serial_key;');
+				$dbh->do('ALTER TABLE ONLY monitoring DROP CONSTRAINT IF EXISTS monitoring_name_key;');
+				$dbh->do('ALTER TABLE ONLY system     DROP CONSTRAINT IF EXISTS system_name_key;');
+				$dbh->do('ALTER TABLE ONLY users      DROP CONSTRAINT IF EXISTS users_usrname_key;');
+
+				$dbh->do('CREATE INDEX cables_cblopt            ON cables       USING btree (cblopt);');
+				$dbh->do('CREATE INDEX cables_jack              ON cables       USING btree (jack);');
+				$dbh->do('CREATE INDEX cables_nbrjack           ON cables       USING btree (nbrjack);');
+				$dbh->do('CREATE INDEX cables_nbrpanel          ON cables       USING btree (nbrpanel);');
+				$dbh->do('CREATE INDEX cables_panel             ON cables       USING btree (panel);');
+				$dbh->do('CREATE INDEX interfaces_device_ifname ON interfaces   USING btree (device, ifname);');
+				$dbh->do('CREATE INDEX iptrack_arpdevice        ON iptrack      USING btree (arpdevice);');
+				$dbh->do('CREATE INDEX iptrack_arpifname        ON iptrack      USING btree (arpifname);');
+				$dbh->do('CREATE INDEX monimap_usrname          ON monimap      USING btree (usrname);');
+				$dbh->do('CREATE INDEX nodes_ifname             ON nodes        USING btree (ifname);');
+				$dbh->do('CREATE INDEX nodes_noduser            ON nodes        USING btree (noduser);');
+				$dbh->do('CREATE INDEX translations_srctype     ON translations USING btree (srctype);');
+				$dbh->do('CREATE INDEX vlanport_device          ON vlanport     USING btree (device);');
+				$dbh->do('CREATE INDEX vlanport_ifname          ON vlanport     USING btree (ifname);');
+				$dbh->do('CREATE INDEX vlanport_vlanid          ON vlanport     USING btree (vlanid);');
+				$dbh->do('CREATE INDEX vlanport_vlopts          ON vlanport     USING btree (vlopts);');
+
+				$dbh->do("ALTER TABLE devices  ALTER vendor   TYPE $vchar(32);");
+				$dbh->do("ALTER TABLE policies ALTER type     TYPE $vchar(128);");
+				$dbh->do("ALTER TABLE vlans    ALTER vlanname TYPE $vchar(64);");
 			}else{
-				$dbh->do("ALTER TABLE policies CHANGE COLUMN type type $vchar(128) DEFAULT '';");
+				## We intentionally add the PRIMARY KEYs here before dropping old INDEXes,
+				## because otherwie MySQL complains about the monimap.id column with:
+				## "Incorrect table definition; there can be only one auto column and it must be defined as a key"
+
+				$dbh->do("ALTER TABLE install    ADD PRIMARY KEY(name);");
+				$dbh->do("ALTER TABLE inventory  ADD PRIMARY KEY(serial);");
+				$dbh->do("ALTER TABLE monimap    ADD PRIMARY KEY(id);");
+				$dbh->do("ALTER TABLE monitoring ADD PRIMARY KEY(name);");
+				$dbh->do("ALTER TABLE system     ADD PRIMARY KEY(name);");
+
+				$dbh->do("DROP INDEX device   ON configs;");
+				$dbh->do("DROP INDEX device_2 ON configs;");
+				$dbh->do("DROP INDEX device   ON devices;");
+				$dbh->do("DROP INDEX device_2 ON devices;");
+				$dbh->do("DROP INDEX id       ON events;");
+				$dbh->do("DROP INDEX id       ON incidents;");
+				$dbh->do("DROP INDEX name     ON install;");
+				$dbh->do("DROP INDEX name_2   ON install;");
+				$dbh->do("DROP INDEX device   ON interfaces;");
+				$dbh->do("DROP INDEX serial   ON inventory;");
+				$dbh->do("DROP INDEX serial_2 ON inventory;");
+				$dbh->do("DROP INDEX id       ON links;");
+				$dbh->do("DROP INDEX id       ON monimap;");
+				$dbh->do("DROP INDEX name     ON monitoring;");
+				$dbh->do("DROP INDEX name_2   ON monitoring;");
+				$dbh->do("DROP INDEX id       ON policies;");
+				$dbh->do("DROP INDEX name     ON system;");
+				$dbh->do("DROP INDEX name_2   ON system;");
+				$dbh->do("DROP INDEX usrname  ON users;");
+
+				$dbh->do("ALTER TABLE devices    CHANGE COLUMN vendor   vendor   $vchar(32)  DEFAULT '';");
+				$dbh->do("ALTER TABLE interfaces CHANGE COLUMN dinbrc   dinbrc   $bigun      DEFAULT 0;");
+				$dbh->do("ALTER TABLE policies   CHANGE COLUMN type     type     $vchar(128) DEFAULT '';");
+				$dbh->do("ALTER TABLE vlans      CHANGE COLUMN vlanname vlanname $vchar(64)  DEFAULT '';");
 			}
 			$dbh->do("ALTER TABLE devices ADD COLUMN totmac $intun DEFAULT 0;");
 			$dbh->do("ALTER TABLE devices ADD COLUMN totarp $intun DEFAULT 0;");
 			$dbh->do("ALTER TABLE devices ADD COLUMN totnd $intun DEFAULT 0;");
 			$dbh->do("ALTER TABLE devices ADD COLUMN devstatus $tinun DEFAULT 0;");
 			$dbh->do("ALTER TABLE devices ADD COLUMN cpualert $tinun DEFAULT 75;");
-			$dbh->do("ALTER TABLE devices ADD COLUMN memalert $intun DEFAULT 1024;");
+			$dbh->do("ALTER TABLE devices ADD COLUMN memalert $intun DEFAULT 25;");
 			$dbh->do("ALTER TABLE devices ADD COLUMN tempalert $tinun DEFAULT 60;");
 			$dbh->do("ALTER TABLE devices ADD COLUMN supplyalert $intun DEFAULT $misc::supa;");
 			$dbh->do("ALTER TABLE devices ADD COLUMN poewarn $tinun DEFAULT $misc::poew;");
@@ -221,6 +297,7 @@
 			$dbh->do("ALTER TABLE iptrack DROP COLUMN aname;");
 
 			$dbh->do("UPDATE system SET VALUE = '$main::VERSION' where name = 'version'");
+			$dbh->commit;
 			print "\nDB update from 1.6.100 completed\n";
 		}else{
 			print "\nDB version $dbver[0] is not supported, sorry!\n";
@@ -264,9 +341,9 @@
 	}
 
 	print "devices\n";
-	my $index = ($misc::backend eq 'Pg')?'':',INDEX (device),INDEX (location),INDEX (contact),PRIMARY KEY (device)';
+	my $index = ($misc::backend eq 'Pg')?'':', INDEX(location), INDEX(contact), PRIMARY KEY(device)';
 	$dbh->do("CREATE TABLE devices(
-		device $vchar(64) NOT NULL UNIQUE,
+		device $vchar(64) NOT NULL,
 		devip $intun DEFAULT 0,
 		serial $vchar(32) DEFAULT '',
 		type $vchar(32)  DEFAULT '',
@@ -331,7 +408,7 @@
 		$index)" );
 
 	print "interfaces\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(device), INDEX(ifname), INDEX(ifidx), INDEX int_devif (device, ifname)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(ifname), INDEX(ifidx), INDEX int_devif (device, ifname)';
 	$dbh->do("CREATE TABLE interfaces(
 		device $vchar(64) NOT NULL,
 		ifname $vchar(32) NOT NULL,
@@ -381,20 +458,20 @@
 		$index)" );
 
 	print "configs\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(device), PRIMARY KEY(device)';
+	$index = ($misc::backend eq 'Pg')?'':', PRIMARY KEY(device)';
 	$dbh->do("CREATE TABLE configs(
-		device $vchar(64) NOT NULL UNIQUE,
+		device $vchar(64) NOT NULL,
 		config $text,
 		changes $text,
 		time $intun DEFAULT 0
 		$index)" );
 
 	print "install\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(name)';
+	$index = ($misc::backend eq 'Pg')?'':', PRIMARY KEY(name)';
 	$dbh->do("CREATE TABLE install(
 		type $vchar(32)  DEFAULT '',
 		target $vchar(32) DEFAULT '',
-		name $vchar(64) NOT NULL UNIQUE,
+		name $vchar(64) NOT NULL,
 		ipaddr $vchar(15) DEFAULT '',
 		mask $vchar(15) DEFAULT '',
 		gateway $vchar(15) DEFAULT '',
@@ -407,10 +484,10 @@
 		$index)" );
 
 	print "inventory\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(serial)';
+	$index = ($misc::backend eq 'Pg')?'':', PRIMARY KEY(serial)';
 	$dbh->do("CREATE TABLE inventory(
 		state $tinun DEFAULT 0,
-		serial $vchar(32) NOT NULL UNIQUE,
+		serial $vchar(32) NOT NULL,
 		assetclass $tinun DEFAULT 1,
 		assettype $vchar(32) DEFAULT 0,
 		assetnumber $vchar(32) DEFAULT '',
@@ -453,7 +530,7 @@
 		$index)" );
 
 	print "links\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(device), INDEX(ifname), INDEX(neighbor), INDEX(nbrifname), PRIMARY KEY(id)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(device), INDEX(ifname), INDEX(neighbor), INDEX(nbrifname), PRIMARY KEY(id)';
 	$dbh->do("CREATE TABLE links(
 		id $serid,
 		device $vchar(64) NOT NULL,
@@ -493,7 +570,7 @@
 		$index)" );
 
 	print "events\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(source), INDEX(level), INDEX(time), INDEX(class), INDEX(device), PRIMARY KEY(id)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(source), INDEX(level), INDEX(time), INDEX(class), INDEX(device), PRIMARY KEY(id)';
 	$dbh->do("CREATE TABLE events(
 		id $serid,
 		level $tinun DEFAULT 0,
@@ -505,9 +582,9 @@
 		$index)" );
 
 	print "monitoring\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(name), INDEX(device)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(device), PRIMARY KEY(name)';
 	$dbh->do("CREATE TABLE monitoring(
-		name $vchar(64) NOT NULL UNIQUE,
+		name $vchar(64) NOT NULL,
 		monip $intun,
 		class $char(4) DEFAULT 'dev',
 		test $char(6) DEFAULT '',
@@ -535,7 +612,7 @@
 		$index)" );
 
 	print "incidents\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(name), INDEX(device), PRIMARY KEY(id)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(name), INDEX(device), PRIMARY KEY(id)';
 	$dbh->do("CREATE TABLE incidents(
 		id $serid,
 		level $tinun DEFAULT 0,
@@ -636,7 +713,7 @@
 		$index)" );
 
 	print "policies\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(status), INDEX(class), PRIMARY KEY(id)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(status), INDEX(class), PRIMARY KEY(id)';
 	$dbh->do("CREATE TABLE policies(
 		id $serid,
 		status $tinun DEFAULT 0,
@@ -674,7 +751,7 @@
 		$index)" );
 
 	print "monimap\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(usrname)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(usrname), PRIMARY KEY(id)';
 	$dbh->do("CREATE TABLE monimap(
 		id $serid,
 		title $vchar(32) DEFAULT '',
@@ -685,7 +762,7 @@
 		$index)" );
 
 	print "cables\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(panel), INDEX(jack), INDEX(nbrpanel), INDEX(nbrjack), INDEX(cblopt), PRIMARY KEY(id)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(panel), INDEX(jack), INDEX(nbrpanel), INDEX(nbrjack), INDEX(cblopt), PRIMARY KEY(id)';
 	$dbh->do("CREATE TABLE cables(
 		id $serid,
 		panel $vchar(64) NOT NULL,
@@ -701,7 +778,7 @@
 # TODO?		tropts $char(4) DEFAULT '',
 # OR		trord  $tinun DEFAULT 0,
 	print "translations\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(id), INDEX(srctype)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(srctype), PRIMARY KEY(id)';
 	$dbh->do("CREATE TABLE translations(
 		id $serid,
 		srctype $vchar(32)  DEFAULT '',
@@ -717,7 +794,7 @@
 	print "users\n";
 	$index = ($misc::backend eq 'Pg')?'':', PRIMARY KEY(usrname)';
 	$dbh->do("CREATE TABLE users(
-		usrname $vchar(32) NOT NULL UNIQUE,
+		usrname $vchar(32) NOT NULL,
 		password $vchar(64) NOT NULL DEFAULT '',
 		groups $smaun NOT NULL DEFAULT '0',
 		email $vchar(64) DEFAULT '',
@@ -738,9 +815,9 @@
 	$sth->execute ( 'admin','3cac26b5bd6addd1ba4f9c96a58ff8c2c2c8ac15018f61240f150a4a968b8562','255',$main::now,'default admin','75','8','10','35' );
 
 	print "system\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(name)';
+	$index = ($misc::backend eq 'Pg')?'':', PRIMARY KEY(name)';
 	$dbh->do("CREATE TABLE system (
-		name $vchar(32) NOT NULL UNIQUE,
+		name $vchar(32) NOT NULL,
 		value $vchar(32) DEFAULT ''
 		$index)" );
 	$sth = $dbh->prepare("INSERT INTO system (name,value) VALUES ( ?,? )");
@@ -749,7 +826,7 @@
 	$sth->execute ( 'version', $main::VERSION );
 
 	print "chat\n";
-	$index = ($misc::backend eq 'Pg')?'':', INDEX(time), INDEX (usrname)';
+	$index = ($misc::backend eq 'Pg')?'':', INDEX(time), INDEX(usrname)';
 	$dbh->do("CREATE TABLE chat (
 		time $intun,
 		usrname $vchar(32) DEFAULT '',
@@ -784,96 +861,118 @@
 		$dbh->do($inet_ntoa);
 #		$dbh->do("ALTER FUNCTION inet_ntoa(bigint) OWNER TO nedi;");TODO remove, if not needed
 
-		$dbh->do('ALTER TABLE ONLY events    ADD CONSTRAINT events_pkey    PRIMARY KEY (id);');
-		$dbh->do('ALTER TABLE ONLY incidents ADD CONSTRAINT incidents_pkey PRIMARY KEY (id);');
-		$dbh->do('ALTER TABLE ONLY links     ADD CONSTRAINT links_pkey     PRIMARY KEY (id);');
-		$dbh->do('ALTER TABLE ONLY locations ADD CONSTRAINT locations_pkey PRIMARY KEY (id);');
-		$dbh->do('ALTER TABLE ONLY nbrtrack  ADD CONSTRAINT nbrtrack_pkey  PRIMARY KEY (id);');
-		$dbh->do('ALTER TABLE ONLY policies  ADD CONSTRAINT policies_pkey  PRIMARY KEY (id);');
-		$dbh->do('ALTER TABLE ONLY install   ADD CONSTRAINT install_pkey   PRIMARY KEY (name);');
-
-		$dbh->do('CREATE INDEX chat_time         ON chat       USING btree (time);');
-		$dbh->do('CREATE INDEX chat_usrname      ON chat       USING btree (usrname);');
-		$dbh->do('CREATE INDEX configs_device    ON configs    USING btree (device);');
-		$dbh->do('CREATE INDEX devices_device    ON devices    USING btree (device);');
-		$dbh->do('CREATE INDEX monimap_usr       ON monimap    USING btree (usrname);');
-		$dbh->do('CREATE INDEX devices_location  ON devices    USING btree (location);');
-		$dbh->do('CREATE INDEX devices_contact   ON devices    USING btree (contact);');
-		$dbh->do('CREATE INDEX dns_aname         ON dns        USING btree (aname);');
-		$dbh->do('CREATE INDEX dns_dnsupdate     ON dns        USING btree (dnsupdate);');
-		$dbh->do('CREATE INDEX dns_nodip         ON dns        USING btree (nodip);');
-		$dbh->do('CREATE INDEX dns6_aaaaname     ON dns6       USING btree (aaaaname);');
-		$dbh->do('CREATE INDEX dns6_dns6update   ON dns6       USING btree (dns6update);');
-		$dbh->do('CREATE INDEX dns6_nodip6       ON dns6       USING btree (nodip6);');
-		$dbh->do('CREATE INDEX events_class      ON events     USING btree (class);');
-		$dbh->do('CREATE INDEX events_device     ON events     USING btree (device);');
-		$dbh->do('CREATE INDEX events_level      ON events     USING btree (level);');
-		$dbh->do('CREATE INDEX events_source     ON events     USING btree (source);');
-		$dbh->do('CREATE INDEX events_time       ON events     USING btree (time);');
-		$dbh->do('CREATE INDEX iftrack_device    ON iftrack    USING btree (device);');
-		$dbh->do('CREATE INDEX iftrack_mac       ON iftrack    USING btree (mac);');
-		$dbh->do('CREATE INDEX iftrack_vlanid    ON iftrack    USING btree (vlanid);');
-		$dbh->do('CREATE INDEX incidents_device  ON incidents  USING btree (device);');
-		$dbh->do('CREATE INDEX incidents_name    ON incidents  USING btree (name);');
-		$dbh->do('CREATE INDEX inventory_serial  ON inventory  USING btree (serial);');
-		$dbh->do('CREATE INDEX install_name      ON install    USING btree (name);');
-		$dbh->do('CREATE INDEX interfaces_device ON interfaces USING btree (device);');
-		$dbh->do('CREATE INDEX interfaces_ifname ON interfaces USING btree (ifname);');
-		$dbh->do('CREATE INDEX interfaces_ifidx  ON interfaces USING btree (ifidx);');
-		$dbh->do('CREATE INDEX interfaces_devif  ON interfaces USING btree (device,ifname);');
-		$dbh->do('CREATE INDEX iptrack_dev       ON iptrack    USING btree (arpdevice);');
-		$dbh->do('CREATE INDEX iptrack_mac       ON iptrack    USING btree (mac);');
-		$dbh->do('CREATE INDEX iptrack_if        ON iptrack    USING btree (arpifname);');
-		$dbh->do('CREATE INDEX links_device      ON links      USING btree (device);');
-		$dbh->do('CREATE INDEX links_ifname      ON links      USING btree (ifname);');
-		$dbh->do('CREATE INDEX links_nbrifname   ON links      USING btree (nbrifname);');
-		$dbh->do('CREATE INDEX links_neighbor    ON links      USING btree (neighbor);');
-		$dbh->do('CREATE INDEX nbrtrack_device   ON nbrtrack   USING btree (device);');
-		$dbh->do('CREATE INDEX nbrtrack_ifname   ON nbrtrack   USING btree (ifname);');
-		$dbh->do('CREATE INDEX nbrtrack_neighbor ON nbrtrack   USING btree (neighbor);');
-		$dbh->do('CREATE INDEX nbrtrack_time     ON nbrtrack   USING btree (time );');
-		$dbh->do('CREATE INDEX locations_region  ON locations  USING btree (region);');
-		$dbh->do('CREATE INDEX modules_device    ON modules    USING btree (device);');
-		$dbh->do('CREATE INDEX modules_serial    ON modules    USING btree (serial);');
-		$dbh->do('CREATE INDEX monitoring_name   ON monitoring USING btree (name);');
-		$dbh->do('CREATE INDEX monitoring_device ON monitoring USING btree (device);');
-		$dbh->do('CREATE INDEX networks_device   ON networks   USING btree (device);');
-		$dbh->do('CREATE INDEX networks_ifip     ON networks   USING btree (ifip);');
-		$dbh->do('CREATE INDEX networks_ifname   ON networks   USING btree (ifname);');
-		$dbh->do('CREATE INDEX netinfo_netip     ON netinfo    USING btree (netip);');
-		$dbh->do('CREATE INDEX netinfo_country   ON netinfo    USING btree (country);');
-		$dbh->do('CREATE INDEX nodarp_arpdevice  ON nodarp     USING btree (arpdevice);');
-		$dbh->do('CREATE INDEX nodarp_arpifname  ON nodarp     USING btree (arpifname);');
-		$dbh->do('CREATE INDEX nodarp_ipupdate   ON nodarp     USING btree (ipupdate);');
-		$dbh->do('CREATE INDEX nodarp_nodip      ON nodarp     USING btree (nodip);');
-		$dbh->do('CREATE INDEX nodarp_mac        ON nodarp     USING btree (mac);');
-		$dbh->do('CREATE INDEX nodarp_srvupdate  ON nodarp     USING btree (srvupdate);');
-		$dbh->do('CREATE INDEX nodnd_nddevice    ON nodnd      USING btree (nddevice);');
-		$dbh->do('CREATE INDEX nodnd_ndifname    ON nodnd      USING btree (ndifname);');
-		$dbh->do('CREATE INDEX nodnd_ip6update   ON nodnd      USING btree (ip6update);');
-		$dbh->do('CREATE INDEX nodnd_nodip6      ON nodnd      USING btree (nodip6);');
-		$dbh->do('CREATE INDEX nodnd_mac         ON nodnd      USING btree (mac);');
-		$dbh->do('CREATE INDEX nodnd_srv6update  ON nodnd      USING btree (srv6update);');
-		$dbh->do('CREATE INDEX nodes_device      ON nodes      USING btree (device);');
-		$dbh->do('CREATE INDEX nodes_mac         ON nodes      USING btree (mac);');
-		$dbh->do('CREATE INDEX nodes_vlanid      ON nodes      USING btree (vlanid);');
-		$dbh->do('CREATE INDEX policies_status   ON policies   USING btree (status);');
-		$dbh->do('CREATE INDEX policies_class    ON policies   USING btree (class);');
-		$dbh->do('CREATE INDEX system_name       ON system     USING btree (name);');
-		$dbh->do('CREATE INDEX vlans_device      ON vlans      USING btree (device);');
-		$dbh->do('CREATE INDEX vlans_vlanid      ON vlans      USING btree (vlanid);');
-		$dbh->do('CREATE INDEX wlan_mac          ON wlan       USING btree (mac);');
-		$dbh->do('CREATE INDEX vlans_device      ON vlanport   USING btree (device);');
-		$dbh->do('CREATE INDEX vlans_vlanid      ON vlanport   USING btree (vlanid);');
-		$dbh->do('CREATE INDEX vlans_ifname      ON vlanport   USING btree (vlanif);');
-		$dbh->do('CREATE INDEX vlans_vlopts      ON vlanport   USING btree (vlopts);');
-		$dbh->do('CREATE INDEX cables_panel      ON cables     USING btree (panel);');
-		$dbh->do('CREATE INDEX cables_jack       ON cables     USING btree (jack);');
-		$dbh->do('CREATE INDEX cables_nbrpanel   ON cables     USING btree (nbrpanel);');
-		$dbh->do('CREATE INDEX cables_nbrjack    ON cables     USING btree (nbrjack);');
-		$dbh->do('CREATE INDEX cables_cblopt     ON cables     USING btree (patopt);');
-		$dbh->do('CREATE INDEX translations_id   ON translations    USING btree (id);');
-		$dbh->do('CREATE INDEX translations_sty  ON translations    USING btree (srctype);');
+		$dbh->do('ALTER TABLE ONLY cables       ADD CONSTRAINT cables_pkey       PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY configs      ADD CONSTRAINT configs_pkey      PRIMARY KEY (device);');
+		$dbh->do('ALTER TABLE ONLY devices      ADD CONSTRAINT devices_pkey      PRIMARY KEY (device);');
+		$dbh->do('ALTER TABLE ONLY events       ADD CONSTRAINT events_pkey       PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY incidents    ADD CONSTRAINT incidents_pkey    PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY install      ADD CONSTRAINT install_pkey      PRIMARY KEY (name);');
+		$dbh->do('ALTER TABLE ONLY inventory    ADD CONSTRAINT inventory_pkey    PRIMARY KEY (serial);');
+		$dbh->do('ALTER TABLE ONLY links        ADD CONSTRAINT links_pkey        PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY locations    ADD CONSTRAINT locations_pkey    PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY monimap      ADD CONSTRAINT monimap_pkey      PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY monitoring   ADD CONSTRAINT monitoring_pkey   PRIMARY KEY (name);');
+		$dbh->do('ALTER TABLE ONLY nbrtrack     ADD CONSTRAINT nbrtrack_pkey     PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY policies     ADD CONSTRAINT policies_pkey     PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY system       ADD CONSTRAINT system_pkey       PRIMARY KEY (name);');
+		$dbh->do('ALTER TABLE ONLY translations ADD CONSTRAINT translations_pkey PRIMARY KEY (id);');
+		$dbh->do('ALTER TABLE ONLY users        ADD CONSTRAINT users_pkey        PRIMARY KEY (usrname);');
+
+		$dbh->do('CREATE INDEX cables_cblopt            ON cables       USING btree (cblopt);');
+		$dbh->do('CREATE INDEX cables_jack              ON cables       USING btree (jack);');
+		$dbh->do('CREATE INDEX cables_nbrjack           ON cables       USING btree (nbrjack);');
+		$dbh->do('CREATE INDEX cables_nbrpanel          ON cables       USING btree (nbrpanel);');
+		$dbh->do('CREATE INDEX cables_panel             ON cables       USING btree (panel);');
+
+		$dbh->do('CREATE INDEX chat_time                ON chat         USING btree (time);');
+		$dbh->do('CREATE INDEX chat_usrname             ON chat         USING btree (usrname);');
+
+		$dbh->do('CREATE INDEX devices_contact          ON devices      USING btree (contact);');
+		$dbh->do('CREATE INDEX devices_location         ON devices      USING btree (location);');
+
+		$dbh->do('CREATE INDEX dns_aname                ON dns          USING btree (aname);');
+		$dbh->do('CREATE INDEX dns_nodip                ON dns          USING btree (nodip);');
+
+		$dbh->do('CREATE INDEX dns6_aaaaname            ON dns6         USING btree (aaaaname);');
+		$dbh->do('CREATE INDEX dns6_nodip6              ON dns6         USING btree (nodip6);');
+
+		$dbh->do('CREATE INDEX events_class             ON events       USING btree (class);');
+		$dbh->do('CREATE INDEX events_device            ON events       USING btree (device);');
+		$dbh->do('CREATE INDEX events_level             ON events       USING btree (level);');
+		$dbh->do('CREATE INDEX events_source            ON events       USING btree (source);');
+		$dbh->do('CREATE INDEX events_time              ON events       USING btree (time);');
+
+		$dbh->do('CREATE INDEX iftrack_device           ON iftrack      USING btree (device);');
+		$dbh->do('CREATE INDEX iftrack_mac              ON iftrack      USING btree (mac);');
+		$dbh->do('CREATE INDEX iftrack_vlanid           ON iftrack      USING btree (vlanid);');
+
+		$dbh->do('CREATE INDEX incidents_device         ON incidents    USING btree (device);');
+		$dbh->do('CREATE INDEX incidents_name           ON incidents    USING btree (name);');
+
+		$dbh->do('CREATE INDEX interfaces_device_ifname ON interfaces   USING btree (device, ifname);');
+		$dbh->do('CREATE INDEX interfaces_ifidx         ON interfaces   USING btree (ifidx);');
+		$dbh->do('CREATE INDEX interfaces_ifname        ON interfaces   USING btree (ifname);');
+
+		$dbh->do('CREATE INDEX iptrack_arpdevice        ON iptrack      USING btree (arpdevice);');
+		$dbh->do('CREATE INDEX iptrack_arpifname        ON iptrack      USING btree (arpifname);');
+		$dbh->do('CREATE INDEX iptrack_mac              ON iptrack      USING btree (mac);');
+
+		$dbh->do('CREATE INDEX links_device             ON links        USING btree (device);');
+		$dbh->do('CREATE INDEX links_ifname             ON links        USING btree (ifname);');
+		$dbh->do('CREATE INDEX links_nbrifname          ON links        USING btree (nbrifname);');
+		$dbh->do('CREATE INDEX links_neighbor           ON links        USING btree (neighbor);');
+
+		$dbh->do('CREATE INDEX locations_region         ON locations    USING btree (region);');
+
+		$dbh->do('CREATE INDEX modules_device           ON modules      USING btree (device);');
+		$dbh->do('CREATE INDEX modules_serial           ON modules      USING btree (serial);');
+
+		$dbh->do('CREATE INDEX monimap_usrname          ON monimap      USING btree (usrname);');
+
+		$dbh->do('CREATE INDEX monitoring_device        ON monitoring   USING btree (device);');
+
+		$dbh->do('CREATE INDEX nbrtrack_device          ON nbrtrack     USING btree (device);');
+		$dbh->do('CREATE INDEX nbrtrack_ifname          ON nbrtrack     USING btree (ifname);');
+		$dbh->do('CREATE INDEX nbrtrack_neighbor        ON nbrtrack     USING btree (neighbor);');
+		$dbh->do('CREATE INDEX nbrtrack_time            ON nbrtrack     USING btree (time);');
+
+		$dbh->do('CREATE INDEX netinfo_country          ON netinfo      USING btree (country);');
+		$dbh->do('CREATE INDEX netinfo_netip            ON netinfo      USING btree (netip);');
+
+		$dbh->do('CREATE INDEX networks_device          ON networks     USING btree (device);');
+		$dbh->do('CREATE INDEX networks_ifip            ON networks     USING btree (ifip);');
+		$dbh->do('CREATE INDEX networks_ifname          ON networks     USING btree (ifname);');
+
+		$dbh->do('CREATE INDEX nodarp_arpdevice         ON nodarp       USING btree (arpdevice);');
+		$dbh->do('CREATE INDEX nodarp_arpifname         ON nodarp       USING btree (arpifname);');
+		$dbh->do('CREATE INDEX nodarp_mac               ON nodarp       USING btree (mac);');
+		$dbh->do('CREATE INDEX nodarp_nodip             ON nodarp       USING btree (nodip);');
+
+		$dbh->do('CREATE INDEX nodes_device             ON nodes        USING btree (device);');
+		$dbh->do('CREATE INDEX nodes_ifname             ON nodes        USING btree (ifname);');
+		$dbh->do('CREATE INDEX nodes_mac                ON nodes        USING btree (mac);');
+		$dbh->do('CREATE INDEX nodes_noduser            ON nodes        USING btree (noduser);');
+		$dbh->do('CREATE INDEX nodes_vlanid             ON nodes        USING btree (vlanid);');
+
+		$dbh->do('CREATE INDEX nodnd_mac                ON nodnd        USING btree (mac);');
+		$dbh->do('CREATE INDEX nodnd_nddevice           ON nodnd        USING btree (nddevice);');
+		$dbh->do('CREATE INDEX nodnd_ndifname           ON nodnd        USING btree (ndifname);');
+		$dbh->do('CREATE INDEX nodnd_nodip6             ON nodnd        USING btree (nodip6);');
+
+		$dbh->do('CREATE INDEX policies_class           ON policies     USING btree (class);');
+		$dbh->do('CREATE INDEX policies_status          ON policies     USING btree (status);');
+
+		$dbh->do('CREATE INDEX translations_srctype     ON translations USING btree (srctype);');
+
+		$dbh->do('CREATE INDEX vlanport_device          ON vlanport     USING btree (device);');
+		$dbh->do('CREATE INDEX vlanport_ifname          ON vlanport     USING btree (ifname);');
+		$dbh->do('CREATE INDEX vlanport_vlanid          ON vlanport     USING btree (vlanid);');
+		$dbh->do('CREATE INDEX vlanport_vlopts          ON vlanport     USING btree (vlopts);');
+
+		$dbh->do('CREATE INDEX vlans_device             ON vlans        USING btree (device);');
+		$dbh->do('CREATE INDEX vlans_vlanid             ON vlans        USING btree (vlanid);');
+
+		$dbh->do('CREATE INDEX wlan_mac                 ON wlan         USING btree (mac);');
 	}
  	$dbh->commit;
 
