--- Nmap/Scanner/Address.pm.orig	2005-08-17 17:50:11.000000000 -0700
+++ Nmap/Scanner/Address.pm	2010-12-30 12:25:56.000000000 -0800
@@ -1,6 +1,7 @@
 package Nmap::Scanner::Address;
 
 use strict;
+use HTML::Entities;
 
 =pod
 
@@ -54,7 +55,7 @@
     return
         '<address addr="' . $self->addr() .
         '" addrtype="' . $self->addrtype() .
-        '" vendor="' . $self->vendor() . '" />';
+        '" vendor="' . encode_entities($self->vendor()) . '" />';
 
 }
 
--- Nmap/Scanner/OS/Class.pm.orig	2005-04-26 08:35:35.000000000 -0700
+++ Nmap/Scanner/OS/Class.pm	2010-12-30 11:54:09.000000000 -0800
@@ -13,6 +13,7 @@
 =cut
 
 use strict;
+use HTML::Entities;
 
 sub new {
     my $class = shift;
@@ -85,7 +86,7 @@
 
     my $xml  = "<osclass";
        $xml .= ' type="'  . $self->type()  . '" ';
-       $xml .= ' vendor="'  . $self->vendor()  . '" ';
+       $xml .= ' vendor="'  . encode_entities($self->vendor())  . '" ';
        $xml .= ' osfamily="'  . $self->osfamily()  . '" ';
        $xml .= ' osgen="'  . $self->osgen()  . '" ' if $self->osgen();
        $xml .= ' accuracy="' . $self->accuracy() . '" ';
