--- config/rstools/config.php.orig	2015-06-22 10:35:32.649915691 -0600
+++ config/rstools/config.php	2015-06-22 10:35:32.656915794 -0600
@@ -61,5 +61,14 @@
    * @var string
    */
   const WEEK_START = 0;
+  
+  /**
+   * GroundWork Roles
+   * 
+   * @var string
+   */
+  const GROUNDWORK_ADMIN = "gw-monitoring-administrator";
+  const GROUNDWORK_OPERATOR = "gw-monitoring-operator";
+  const GROUNDWORK_USER = "gw-monitoring-user";
 }
 ?>
\ No newline at end of file
