--- foundation/container/jpp/standalone/configuration/check-listener.conf.orig	2015-06-22 10:35:32.756917265 -0600
+++ foundation/container/jpp/standalone/configuration/check-listener.conf	2015-06-22 10:35:32.758917294 -0600
@@ -46,6 +46,7 @@
 secondary_deployment = /usr/local/groundwork/foundation/container/jpp/standalone/deployments/nms-rstools.war
 
 # Secondary deployments that we thought we wouldn't have to deal with here.
+secondary_deployment = /usr/local/groundwork/foundation/container/jpp/standalone/deployments/gwos-jbossas7-monitoringAgent.war
 secondary_deployment = /usr/local/groundwork/foundation/container/jpp/standalone/deployments/monarch-export.war
 secondary_deployment = /usr/local/groundwork/foundation/container/jpp/standalone/deployments/nagios-app.war
 secondary_deployment = /usr/local/groundwork/foundation/container/jpp/standalone/deployments/nagvis_share.war
