--- nagios-plugins-1.4.13/plugins-scripts/check_ntp.pl	2005-11-29 16:49:47.000000000 -0800
+++ nagios-plugins-1.4.13_mod/plugins-scripts/check_ntp.pl	2009-06-03 04:00:57.000000000 -0700
@@ -62,7 +62,6 @@
 use strict;
 use Getopt::Long;
 use vars qw($opt_V $opt_h $opt_H $opt_t $opt_w $opt_c $opt_O $opt_j $opt_k $verbose $PROGNAME $def_jitter $ipv4 $ipv6);
-use lib utils.pm;
 use utils qw($TIMEOUT %ERRORS &print_revision &support);
 
 $PROGNAME="check_ntp";
