--- nagios-plugins-1.4.13/configure.in	2008-09-25 01:15:58.000000000 -0700
+++ nagios-plugins-1.4.13_mod/configure.in	2009-06-03 02:03:15.000000000 -0700
@@ -1172,6 +1172,12 @@
                             [sets path to rpcinfo]), PATH_TO_RPCINFO=$withval)
 AC_DEFINE_UNQUOTED(PATH_TO_RPCINFO,"$PATH_TO_RPCINFO",[path to rpcinfo binary])
 
+AC_PATH_PROG(PATH_TO_NTPDATE,ntpdate)
+AC_ARG_WITH(ntpdate_command,
+            ACX_HELP_STRING([--with-ntpdate-command=PATH],
+                            [sets path to ntpdate]), PATH_TO_NTPDATE=$withval)
+AC_DEFINE_UNQUOTED(PATH_TO_NTPDATE,"$PATH_TO_NTPDATE",[path to ntpdate binary])
+
 AC_PATH_PROG(PATH_TO_LMSTAT,lmstat)
 if test -x "$PATH_TO_LMSTAT"
 then
