--- ServiceNow-1.01/lib/ServiceNow.pm.orig	2010-03-04 14:33:56.000000000 -0800
+++ ServiceNow-1.01/lib/ServiceNow.pm	2011-05-19 10:08:47.000000000 -0700
@@ -23,7 +23,7 @@
 
 =head1 ServiceNow module
 
-Service-now Perl API - ServiceNow perl module
+ServiceNow - ServiceNow Perl API module
 
 =head1 Desciption
 
@@ -767,4 +767,4 @@
 	return $target->queryFields($table,$choices); 
 }
     
-1;
\ No newline at end of file
+1;
