*** cacti.orig/htdocs/data_sources.php	Fri Oct  5 02:24:47 2012
--- cacti/htdocs/data_sources.php	Sat Nov  3 23:25:23 2012
***************
*** 1034,1039 ****
--- 1034,1054 ----
  	}
  }
  
+ function filter_highlight_and_encode($target, $filter) {
+ 	if (trim($filter) == '') {
+ 		return htmlspecialchars($target, ENT_QUOTES);
+ 	}
+ 	$preg_filter = '/(' . preg_quote($filter, '/') . ')/i';
+ 	$segments = preg_split($preg_filter, $target, NULL, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
+ 	$out = array();
+ 	foreach ($segments as $segment) {
+ 		$out[] = (strcasecmp($segment, $filter) == 0)
+ 			? ("<span style='background-color: #F8D93D;'>". htmlspecialchars($segment, ENT_QUOTES) . '</span>')
+ 			: htmlspecialchars($segment, ENT_QUOTES);
+ 	}
+ 	return implode('',$out);
+ }
+ 
  function ds() {
  	global $colors, $ds_actions, $item_rows, $database_type;
  
***************
*** 1356,1373 ****
  	$i = 0;
  	if (sizeof($data_sources) > 0) {
  		foreach ($data_sources as $data_source) {
- 			$data_source = api_plugin_hook_function('data_sources_table', $data_source);
  			/* we're escaping strings here, so no need to escape them on form_selectable_cell */
! 			$data_template_name = ((empty($data_source["data_template_name"])) ? "<em>None</em>" : htmlspecialchars($data_source["data_template_name"]));
! 			$data_input_name    = ((empty($data_source["data_input_name"])) ? "<em>External</em>" : htmlspecialchars($data_source["data_input_name"]));
  			$poller_interval    = ((isset($poller_intervals[$data_source["local_data_id"]])) ? $poller_intervals[$data_source["local_data_id"]] : 0);
  			form_alternate_row_color($colors["alternate"], $colors["light"], $i, 'line' . $data_source["local_data_id"]); $i++;
! 			form_selectable_cell("<a class='linkEditMain' href='" . htmlspecialchars("data_sources.php?action=ds_edit&id=" . $data_source["local_data_id"]) . "' title='" . $data_source["name_cache"] . "'>" . ((get_request_var_request("filter") != "") ? eregi_replace("(" . preg_quote(get_request_var_request("filter")) . ")", "<span style='background-color: #F8D93D;'>\\1</span>", title_trim(htmlspecialchars($data_source["name_cache"]), read_config_option("max_title_data_source"))) : title_trim(htmlspecialchars($data_source["name_cache"]), read_config_option("max_title_data_source"))) . "</a>", $data_source["local_data_id"]);
  			form_selectable_cell($data_source['local_data_id'], $data_source['local_data_id']);
! 			form_selectable_cell(((get_request_var_request("filter") != "") ? eregi_replace("(" . preg_quote(get_request_var_request("filter")) . ")", "<span style='background-color: #F8D93D;'>\\1</span>", $data_input_name) : $data_input_name), $data_source["local_data_id"]);
  			form_selectable_cell(get_poller_interval($poller_interval), $data_source["local_data_id"]);
  			form_selectable_cell(($data_source['active'] == "on" ? "Yes" : "No"), $data_source["local_data_id"]);
! 			form_selectable_cell(((get_request_var_request("filter") != "") ? eregi_replace("(" . preg_quote(get_request_var_request("filter")) . ")", "<span style='background-color: #F8D93D;'>\\1</span>", $data_template_name) : $data_template_name), $data_source["local_data_id"]);
  			form_checkbox_cell($data_source["name_cache"], $data_source["local_data_id"]);
  			form_end_row();
  		}
--- 1371,1391 ----
  	$i = 0;
  	if (sizeof($data_sources) > 0) {
  		foreach ($data_sources as $data_source) {
  			/* we're escaping strings here, so no need to escape them on form_selectable_cell */
! 			$data_source["data_template_name"] = filter_highlight_and_encode($data_source["data_template_name"], get_request_var_request("filter"));
! 			$data_source["data_input_name"]    = filter_highlight_and_encode($data_source["data_input_name"],    get_request_var_request("filter"));
! 			$data_name_cache                   = filter_highlight_and_encode(title_trim($data_source["name_cache"], read_config_option("max_title_data_source")), get_request_var_request("filter"));
! 			$data_source = api_plugin_hook_function('data_sources_table', $data_source);
! 			$data_template_name = ((empty($data_source["data_template_name"])) ? "<em>None</em>"     : $data_source["data_template_name"]);
! 			$data_input_name    = ((empty($data_source["data_input_name"]))    ? "<em>External</em>" : $data_source["data_input_name"]);
  			$poller_interval    = ((isset($poller_intervals[$data_source["local_data_id"]])) ? $poller_intervals[$data_source["local_data_id"]] : 0);
  			form_alternate_row_color($colors["alternate"], $colors["light"], $i, 'line' . $data_source["local_data_id"]); $i++;
! 			form_selectable_cell("<a class='linkEditMain' href='" . htmlspecialchars("data_sources.php?action=ds_edit&id=" . $data_source["local_data_id"]) . "' title='" . htmlspecialchars($data_source["name_cache"], ENT_QUOTES) . "'>" . $data_name_cache . "</a>", $data_source["local_data_id"]);
  			form_selectable_cell($data_source['local_data_id'], $data_source['local_data_id']);
! 			form_selectable_cell($data_input_name, $data_source["local_data_id"]);
  			form_selectable_cell(get_poller_interval($poller_interval), $data_source["local_data_id"]);
  			form_selectable_cell(($data_source['active'] == "on" ? "Yes" : "No"), $data_source["local_data_id"]);
! 			form_selectable_cell($data_template_name, $data_source["local_data_id"]);
  			form_checkbox_cell($data_source["name_cache"], $data_source["local_data_id"]);
  			form_end_row();
  		}
