*** html.php	2015-12-04 15:27:26.440507676 -0500
--- html.php.fixed	2015-12-04 15:27:23.127657197 -0500
***************
*** 383,389 ****
  			print "<td " . ((($i+1) == count($header_items)) ? "colspan='$last_item_colspan' " : "") . "class='textSubHeaderDark'>" . $display_text . "</td>\n";
  		}else{
  			print "<td " . ((($i) == count($header_items)) ? "colspan='$last_item_colspan'>" : ">");
! 			print "<a class='textSubHeaderDark' href='" . htmlspecialchars($config['url_path'] . $_SERVER["PHP_SELF"] . "?sort_column=" . $db_column . "&sort_direction=" . $direction) . "'>" . $display_text . "</a>";
  			print "</td>\n";
  		}
  
--- 383,390 ----
  			print "<td " . ((($i+1) == count($header_items)) ? "colspan='$last_item_colspan' " : "") . "class='textSubHeaderDark'>" . $display_text . "</td>\n";
  		}else{
  			print "<td " . ((($i) == count($header_items)) ? "colspan='$last_item_colspan'>" : ">");
! 			#print "<a class='textSubHeaderDark' href='" . htmlspecialchars($config['url_path'] . $_SERVER["PHP_SELF"] . "?sort_column=" . $db_column . "&sort_direction=" . $direction) . "'>" . $display_text . "</a>";
! 			print "<a class='textSubHeaderDark' href='" . htmlspecialchars( $_SERVER["PHP_SELF"] . "?sort_column=" . $db_column . "&sort_direction=" . $direction) . "'>" . $display_text . "</a>";
  			print "</td>\n";
  		}
  
***************
*** 433,439 ****
  			print "<td class='textSubHeaderDark'>" . $display_text . "</td>\n";
  		}else{
  			print "<td>";
! 			print "<a class='textSubHeaderDark' href='" . htmlspecialchars($config['url_path'] . $_SERVER["PHP_SELF"] . "?sort_column=" . $db_column . "&sort_direction=" . $direction) . "'>" . $display_text . "</a>";
  			print "</td>\n";
  		}
  	}
--- 434,441 ----
  			print "<td class='textSubHeaderDark'>" . $display_text . "</td>\n";
  		}else{
  			print "<td>";
! 			#print "<a class='textSubHeaderDark' href='" . htmlspecialchars($config['url_path'] . $_SERVER["PHP_SELF"] . "?sort_column=" . $db_column . "&sort_direction=" . $direction) . "'>" . $display_text . "</a>";
! 			print "<a class='textSubHeaderDark' href='" . htmlspecialchars( $_SERVER["PHP_SELF"] . "?sort_column=" . $db_column . "&sort_direction=" . $direction) . "'>" . $display_text . "</a>";
  			print "</td>\n";
  		}
  	}
