--- cacti/htdocs/lib/snmp.php	2010-07-09 15:33:46.000000000 -0700
+++ cacti/htdocs-security/lib/snmp.php	2013-10-30 11:17:47.519726081 -0700
@@ -84,9 +84,9 @@
 		$timeout = ceil($timeout / 1000);
 
 		if ($version == "1") {
-			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? snmp_escape_string($community): "-c " . snmp_escape_string($community); /* v1/v2 - community string */
+			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? escapeshellarg($community): "-c " . escapeshellarg($community); /* v1/v2 - community string */
 		}elseif ($version == "2") {
-			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? snmp_escape_string($community) : "-c " . snmp_escape_string($community); /* v1/v2 - community string */
+			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? escapeshellarg($community) : "-c " . escapeshellarg($community); /* v1/v2 - community string */
 			$version = "2c"; /* ucd/net snmp prefers this over '2' */
 		}elseif ($version == "3") {
 			if ($priv_proto == "[None]") {
@@ -97,21 +97,21 @@
 			}
 
 			if (strlen($priv_pass)) {
-				$priv_pass = "-X " . snmp_escape_string($priv_pass) . " -x " . snmp_escape_string($priv_proto);
+				$priv_pass = "-X " . escapeshellarg($priv_pass) . " -x " . escapeshellarg($priv_proto);
 			}else{
 				$priv_pass = "";
 			}
 
 			if (strlen($context)) {
-				$context = "-n " . snmp_escape_string($context);
+				$context = "-n " . escapeshellarg($context);
 			}else{
 				$context = "";
 			}
 
-			$snmp_auth = trim("-u " . snmp_escape_string($username) .
-				" -l " . snmp_escape_string($proto) .
-				" -a " . snmp_escape_string($auth_proto) .
-				" -A " . snmp_escape_string($password) .
+			$snmp_auth = trim("-u " . escapeshellarg($username) .
+				" -l " . escapeshellarg($proto) .
+				" -a " . escapeshellarg($auth_proto) .
+				" -A " . escapeshellarg($password) .
 				" "    . $priv_pass .
 				" "    . $context); /* v3 - username/password */
 		}
@@ -199,9 +199,9 @@
 		$timeout = ceil($timeout / 1000);
 
 		if ($version == "1") {
-			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? snmp_escape_string($community): "-c " . snmp_escape_string($community); /* v1/v2 - community string */
+			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? escapeshellarg($community): "-c " . escapeshellarg($community); /* v1/v2 - community string */
 		}elseif ($version == "2") {
-			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? snmp_escape_string($community): "-c " . snmp_escape_string($community); /* v1/v2 - community string */
+			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? escapeshellarg($community): "-c " . escapeshellarg($community); /* v1/v2 - community string */
 			$version = "2c"; /* ucd/net snmp prefers this over '2' */
 		}elseif ($version == "3") {
 			if ($priv_proto == "[None]") {
@@ -212,21 +212,21 @@
 			}
 
 			if (strlen($priv_pass)) {
-				$priv_pass = "-X " . snmp_escape_string($priv_pass) . " -x " . snmp_escape_string($priv_proto);
+				$priv_pass = "-X " . escapeshellarg($priv_pass) . " -x " . escapeshellarg($priv_proto);
 			}else{
 				$priv_pass = "";
 			}
 
 			if (strlen($context)) {
-				$context = "-n " . snmp_escape_string($context);
+				$context = "-n " . escapeshellarg($context);
 			}else{
 				$context = "";
 			}
 
-			$snmp_auth = trim("-u " . snmp_escape_string($username) .
-				" -l " . snmp_escape_string($proto) .
-				" -a " . snmp_escape_string($auth_proto) .
-				" -A " . snmp_escape_string($password) .
+			$snmp_auth = trim("-u " . escapeshellarg($username) .
+				" -l " . escapeshellarg($proto) .
+				" -a " . escapeshellarg($auth_proto) .
+				" -A " . escapeshellarg($password) .
 				" "    . $priv_pass .
 				" "    . $context); /* v3 - username/password */
 		}
@@ -348,9 +348,9 @@
 		$timeout = ceil($timeout / 1000);
 
 		if ($version == "1") {
-			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? snmp_escape_string($community): "-c " . snmp_escape_string($community); /* v1/v2 - community string */
+			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? escapeshellarg($community): "-c " . escapeshellarg($community); /* v1/v2 - community string */
 		}elseif ($version == "2") {
-			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? snmp_escape_string($community): "-c " . snmp_escape_string($community); /* v1/v2 - community string */
+			$snmp_auth = (read_config_option("snmp_version") == "ucd-snmp") ? escapeshellarg($community): "-c " . escapeshellarg($community); /* v1/v2 - community string */
 			$version = "2c"; /* ucd/net snmp prefers this over '2' */
 		}elseif ($version == "3") {
 			if ($priv_proto == "[None]") {
@@ -361,21 +361,21 @@
 			}
 
 			if (strlen($priv_pass)) {
-				$priv_pass = "-X " . snmp_escape_string($priv_pass) . " -x " . snmp_escape_string($priv_proto);
+				$priv_pass = "-X " . escapeshellarg($priv_pass) . " -x " . escapeshellarg($priv_proto);
 			}else{
 				$priv_pass = "";
 			}
 
 			if (strlen($context)) {
-				$context = "-n " . snmp_escape_string($context);
+				$context = "-n " . escapeshellarg($context);
 			}else{
 				$context = "";
 			}
 
-			$snmp_auth = trim("-u " . snmp_escape_string($username) .
-				" -l " . snmp_escape_string($proto) .
-				" -a " . snmp_escape_string($auth_proto) .
-				" -A " . snmp_escape_string($password) .
+			$snmp_auth = trim("-u " . escapeshellarg($username) .
+				" -l " . escapeshellarg($proto) .
+				" -a " . escapeshellarg($auth_proto) .
+				" -A " . escapeshellarg($password) .
 				" "    . $priv_pass .
 				" "    . $context); /* v3 - username/password */
 		}
@@ -551,14 +551,6 @@
 	return $string;
 }
 
-function snmp_escape_string($string) {
-	if (substr_count($string, SNMP_ESCAPE_CHARACTER)) {
-		$string = substr_replace(SNMP_ESCAPE_CHARACTER, "\\" . SNMP_ESCAPE_CHARACTER, $string);
-	}
-
-	return SNMP_ESCAPE_CHARACTER . $string . SNMP_ESCAPE_CHARACTER;
-}
-
 function snmp_get_method($version = 1) {
 	if ((function_exists("snmpget")) && ($version == 1)) {
 		return SNMP_METHOD_PHP;
