--- cacti/htdocs/lib/snmp.php.pre_ipv6	2017-09-26 20:20:44.000000000 -0700
+++ cacti/htdocs/lib/snmp.php	2017-09-27 03:54:12.192861565 -0700
@@ -36,8 +36,16 @@
 	define("SNMP_ESCAPE_CHARACTER", "\"");
 }
 
+function is_raw_ipv6_addr($hostname) {
+	if ( (strlen($hostname) == 0) || (strpos($hostname, ':') === false) || (substr($hostname, 0, 1) === '[') ) {
+		return false;
+	}
+	return true;
+}
+
 function cacti_snmp_get($hostname, $community, $oid, $version, $username, $password, $auth_proto, $priv_pass, $priv_proto, $context, $port = 161, $timeout = 500, $retries = 0, $environ = SNMP_POLLER) {
 	global $config;
+	$portsafe_hostname = is_raw_ipv6_addr($hostname) ? "[$hostname]" : $hostname;
 
 	/* determine default retries */
 	if (($retries == 0) || (!is_numeric($retries))) {
@@ -62,9 +70,9 @@
 		snmp_set_quick_print(0);
 
 		if ($version == "1") {
-			$snmp_value = @snmpget("$hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
+			$snmp_value = @snmpget("$portsafe_hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
 		}elseif ($version == "2") {
-			$snmp_value = @snmp2_get("$hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
+			$snmp_value = @snmp2_get("$portsafe_hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
 		}else{
 			if ($priv_proto == "[None]") {
 				$proto = "authNoPriv";
@@ -73,7 +81,7 @@
 				$proto = "authPriv";
 			}
 
-			$snmp_value = @snmp3_get("$hostname:$port", "$username", $proto, $auth_proto, "$password", $priv_proto, "$priv_pass", "$oid", ($timeout * 1000), $retries);
+			$snmp_value = @snmp3_get("$portsafe_hostname:$port", "$username", $proto, $auth_proto, "$password", $priv_proto, "$priv_pass", "$oid", ($timeout * 1000), $retries);
 		}
 
 		if ($snmp_value === false) {
@@ -123,9 +131,9 @@
 			/* escape the command to be executed and vulnerable parameters
 			 * numeric parameters are not subject to command injection
 			 * snmp_auth is treated seperately, see above */
-			exec(cacti_escapeshellcmd(read_config_option("path_snmpget")) . " -O vt -v$version -t $timeout -r $retries " . cacti_escapeshellarg($hostname) . ":$port $snmp_auth " . cacti_escapeshellarg($oid), $snmp_value);
+			exec(cacti_escapeshellcmd(read_config_option("path_snmpget")) . " -O vt -v$version -t $timeout -r $retries " . cacti_escapeshellarg($portsafe_hostname) . ":$port $snmp_auth " . cacti_escapeshellarg($oid), $snmp_value);
 		}else {
-			exec(cacti_escapeshellcmd(read_config_option("path_snmpget")) . " -O fntev " . $snmp_auth . " -v $version -t $timeout -r $retries " . cacti_escapeshellarg($hostname) . ":$port " . cacti_escapeshellarg($oid), $snmp_value);
+			exec(cacti_escapeshellcmd(read_config_option("path_snmpget")) . " -O fntev " . $snmp_auth . " -v $version -t $timeout -r $retries " . cacti_escapeshellarg($portsafe_hostname) . ":$port " . cacti_escapeshellarg($oid), $snmp_value);
 		}
 
 		/* fix for multi-line snmp output */
@@ -153,6 +161,7 @@
 
 function cacti_snmp_getnext($hostname, $community, $oid, $version, $username, $password, $auth_proto, $priv_pass, $priv_proto, $context, $port = 161, $timeout = 500, $retries = 0, $environ = SNMP_POLLER) {
 	global $config;
+	$portsafe_hostname = is_raw_ipv6_addr($hostname) ? "[$hostname]" : $hostname;
 
 	/* determine default retries */
 	if (($retries == 0) || (!is_numeric($retries))) {
@@ -177,9 +186,9 @@
 		snmp_set_quick_print(0);
 
 		if ($version == "1") {
-			$snmp_value = @snmpgetnext("$hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
+			$snmp_value = @snmpgetnext("$portsafe_hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
 		}elseif ($version == "2") {
-			$snmp_value = @snmp2_getnext("$hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
+			$snmp_value = @snmp2_getnext("$portsafe_hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
 		}else{
 			if ($priv_proto == "[None]") {
 				$proto = "authNoPriv";
@@ -188,7 +197,7 @@
 				$proto = "authPriv";
 			}
 
-			$snmp_value = @snmp3_getnext("$hostname:$port", "$username", $proto, $auth_proto, "$password", $priv_proto, "$priv_pass", "$oid", ($timeout * 1000), $retries);
+			$snmp_value = @snmp3_getnext("$portsafe_hostname:$port", "$username", $proto, $auth_proto, "$password", $priv_proto, "$priv_pass", "$oid", ($timeout * 1000), $retries);
 		}
 
 		if ($snmp_value === false) {
@@ -238,9 +247,9 @@
 			/* escape the command to be executed and vulnerable parameters
 			 * numeric parameters are not subject to command injection
 			 * snmp_auth is treated seperately, see above */
-			exec(cacti_escapeshellcmd(read_config_option("path_snmpgetnext")) . " -O vt -v$version -t $timeout -r $retries " . cacti_escapeshellarg($hostname) . ":$port $snmp_auth " . cacti_escapeshellarg($oid), $snmp_value);
+			exec(cacti_escapeshellcmd(read_config_option("path_snmpgetnext")) . " -O vt -v$version -t $timeout -r $retries " . cacti_escapeshellarg($portsafe_hostname) . ":$port $snmp_auth " . cacti_escapeshellarg($oid), $snmp_value);
 		}else {
-			exec(cacti_escapeshellcmd(read_config_option("path_snmpgetnext")) . " -O fntev $snmp_auth -v $version -t $timeout -r $retries " . cacti_escapeshellarg($hostname) . ":$port " . cacti_escapeshellarg($oid), $snmp_value);
+			exec(cacti_escapeshellcmd(read_config_option("path_snmpgetnext")) . " -O fntev $snmp_auth -v $version -t $timeout -r $retries " . cacti_escapeshellarg($portsafe_hostname) . ":$port " . cacti_escapeshellarg($oid), $snmp_value);
 		}
 	}
 
@@ -263,6 +272,7 @@
 
 function cacti_snmp_walk($hostname, $community, $oid, $version, $username, $password, $auth_proto, $priv_pass, $priv_proto, $context, $port = 161, $timeout = 500, $retries = 0, $max_oids = 10, $environ = SNMP_POLLER) {
 	global $config, $banned_snmp_strings;
+	$portsafe_hostname = is_raw_ipv6_addr($hostname) ? "[$hostname]" : $hostname;
 
 	$snmp_oid_included = false;
 	$snmp_auth	       = '';
@@ -305,9 +315,9 @@
 		snmp_set_quick_print(0);
 
 		if ($version == "1") {
-			$temp_array = @snmprealwalk("$hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
+			$temp_array = @snmprealwalk("$portsafe_hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
 		}elseif ($version == "2") {
-			$temp_array = @snmp2_real_walk("$hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
+			$temp_array = @snmp2_real_walk("$portsafe_hostname:$port", "$community", "$oid", ($timeout * 1000), $retries);
 		}else{
 			if ($priv_proto == "[None]") {
 				$proto = "authNoPriv";
@@ -316,7 +326,7 @@
 				$proto = "authPriv";
 			}
 
-			$temp_array = @snmp3_real_walk("$hostname:$port", "$username", $proto, $auth_proto, "$password", $priv_proto, "$priv_pass", "$oid", ($timeout * 1000), $retries);
+			$temp_array = @snmp3_real_walk("$portsafe_hostname:$port", "$username", $proto, $auth_proto, "$password", $priv_proto, "$priv_pass", "$oid", ($timeout * 1000), $retries);
 		}
 
 		if ($temp_array === false) {
@@ -384,12 +394,12 @@
 			/* escape the command to be executed and vulnerable parameters
 			 * numeric parameters are not subject to command injection
 			 * snmp_auth is treated seperately, see above */
-			$temp_array = exec_into_array(cacti_escapeshellcmd(read_config_option("path_snmpwalk")) . " -v$version -t $timeout -r $retries " . cacti_escapeshellarg($hostname) . ":$port $snmp_auth " . cacti_escapeshellarg($oid));
+			$temp_array = exec_into_array(cacti_escapeshellcmd(read_config_option("path_snmpwalk")) . " -v$version -t $timeout -r $retries " . cacti_escapeshellarg($portsafe_hostname) . ":$port $snmp_auth " . cacti_escapeshellarg($oid));
 		}else {
 			if (file_exists($path_snmpbulkwalk) && ($version > 1) && ($max_oids > 1)) {
-				$temp_array = exec_into_array(cacti_escapeshellcmd($path_snmpbulkwalk) . " -O Qn $snmp_auth -v $version -t $timeout -r $retries -Cr$max_oids " . cacti_escapeshellarg($hostname) . ":$port " . cacti_escapeshellarg($oid));
+				$temp_array = exec_into_array(cacti_escapeshellcmd($path_snmpbulkwalk) . " -O Qn $snmp_auth -v $version -t $timeout -r $retries -Cr$max_oids " . cacti_escapeshellarg($portsafe_hostname) . ":$port " . cacti_escapeshellarg($oid));
 			}else{
-				$temp_array = exec_into_array(cacti_escapeshellcmd(read_config_option("path_snmpwalk")) . " -O Qn $snmp_auth -v $version -t $timeout -r $retries " . cacti_escapeshellarg($hostname) . ":$port " . cacti_escapeshellarg($oid));
+				$temp_array = exec_into_array(cacti_escapeshellcmd(read_config_option("path_snmpwalk")) . " -O Qn $snmp_auth -v $version -t $timeout -r $retries " . cacti_escapeshellarg($portsafe_hostname) . ":$port " . cacti_escapeshellarg($oid));
 			}
 		}
 
