--- cacti/htdocs/plugins/thold/thold_functions.php.orig	2017-05-18 20:16:39.000000000 -0700
+++ cacti/htdocs/plugins/thold/thold_functions.php	2017-05-19 16:37:18.000000000 -0700
@@ -761,7 +761,7 @@
 }
 
 function logger($desc, $breach_up, $threshld, $currentval, $trigger, $triggerct) {
-	define_syslog_variables();
+	// define_syslog_variables();
 
 	$syslog_level = read_config_option('thold_syslog_level');
 	$syslog_facility = read_config_option('thold_syslog_facility');
@@ -1615,7 +1615,7 @@
 			$log_type = "note";
 
 		if (strlen($log_type)) {
-			define_syslog_variables();
+			// define_syslog_variables();
 
 			if ($config["cacti_server_os"] == "win32")
 				openlog("Cacti", LOG_NDELAY | LOG_PID, LOG_USER);
