--- httpd.conf.orig	2009-11-05 11:32:44.000000000 -0800
+++ httpd.conf	2009-11-05 11:34:14.000000000 -0800
@@ -194,17 +194,9 @@
 	
 Alias /nedi "/usr/local/groundwork/nms/applications/nedi/html"
 <Directory "/usr/local/groundwork/nms/applications/nedi/html">
-	AuthType Basic
-	require valid-user
 	DirectoryIndex index.php
 
-	TKTAuthCookieName nedi_auth_tkt
-	TKTAuthDomain groundwork.groundworkopensource.com
-	TKTAuthTimeout 0
-
 	Options FollowSymLinks Indexes
-	AllowOverride None
-	Order allow,deny
 	Allow from all
 </Directory>
 
@@ -215,17 +207,9 @@
 Alias /cacti "/usr/local/groundwork/nms/applications/cacti"
 <Directory "/usr/local/groundwork/nms/applications/cacti">
 
-	AuthType Basic
-	require valid-user
 	DirectoryIndex index.php
 
-	TKTAuthCookieName cacti_auth_tkt
-	TKTAuthDomain groundwork.groundworkopensource.com
-	TKTAuthTimeout 0
-
 	Options FollowSymLinks Indexes
-	AllowOverride None
-	Order allow,deny
 	Allow from all
 </Directory>
 
@@ -525,3 +509,66 @@
 SSLRandomSeed startup builtin
 SSLRandomSeed connect builtin
 </IfModule>
+
+SetEnvIf Referer https?://\S+/portal/ framework_referer
+SetEnvIf Referer https?://\S+/cacti/ / framework_referer
+SetEnvIf Referer https?://\S+/nedi/ / framework_referer
+
+#
+# Deny any access except refers from the portal
+# 
+<Location /cacti/>
+	Order Deny,Allow
+	Deny from all
+	Allow from env=framework_referer
+</Location>
+
+#
+# The following pages refreshes themself. 
+# Refer will not work and therefore pages check for
+# valid session
+#
+<Location /cacti/utilities.php>
+        Order Deny,Allow
+        Allow from all
+</Location>
+
+<Location /cacti/graph_view.php>
+        Order Deny,Allow
+        Allow from all
+</Location>
+
+<Location /cacti/graph.php>
+        Order Deny,Allow
+        Allow from all
+</Location>
+
+<Location /cacti/graph_image.php>
+        Order Deny,Allow
+        Allow from all
+</Location>
+
+<Location  /cacti/plugins/thold/thold_graph.php>
+	Order Deny,Allow
+        Allow from all
+</Location>
+
+<Location /cacti/plugins/weathermap/weathermap-cacti-plugin.php >
+         Order Deny,Allow
+        Allow from all
+</Location>
+
+<Location /cacti/plugins/discovery/discover.php >
+         Order Deny,Allow
+        Allow from all
+</Location>
+
+#
+# Allow refer from portal
+#
+<Location /nedi/>
+	Order Deny,Allow
+	Deny from all
+	Allow from env=framework_referer
+</Location>
+
