*** graph_view.php.orig	2009-11-03 17:03:04.000000000 -0800
--- graph_view.php	2009-11-03 16:37:20.000000000 -0800
***************
*** 39,44 ****
--- 39,49 ----
  input_validate_input_regex(get_request_var_request('graph_remove'), "^([\,0-9]+)$");
  /* ==================================================== */
  
+ if ( strlen( $_SESSION['GWRK_USER']) == 0 ) {
+         print "<br><br><center><font color=red>You do not have permission to access this page. Please log in first</font></color>";
+         exit;
+ }
+ 
  if (isset($_GET["hide"])) {
  	if (($_GET["hide"] == "0") || ($_GET["hide"] == "1")) {
  		/* only update expand/contract info is this user has rights to keep their own settings */
