*** discover.php.orig	2009-11-03 17:01:10.000000000 -0800
--- discover.php	2009-11-03 16:38:21.000000000 -0800
***************
*** 25,30 ****
--- 25,36 ----
  chdir('../../');
  include("./include/auth.php");
  
+ if ( strlen( $_SESSION['GWRK_USER']) == 0 ) {
+         print "<br><br><center><font color=red>You do not have permission to access this page. Please log in first</font></color>";
+         exit;
+ }
+ 
+ 
  discovery_setup_table();
  
  $hosts = array();
