--- nagios-4.3.4/base/config.c.orig	2017-08-24 14:43:48.000000000 -0700
+++ nagios-4.3.4/base/config.c	2017-09-02 07:02:20.166176063 -0700
@@ -29,6 +29,8 @@
 #include "../include/nebmods.h"
 #include "../include/nebmodules.h"
 
+/* Establish that this patch is in place. */
+char no_deprecation_patch_ident[] = "$NoDeprecationWarningsPatchCompileTime: " __TIME__ " on " __DATE__ " (" __FILE__ ") $";
 
 /*** helpers ****/
 /*
@@ -1188,7 +1190,10 @@
 
 	if (deprecated) {
 		for (list = deprecated; list; list = list->next) {
-			logit(NSLOG_CONFIG_WARNING, TRUE, "%s", (char *)list->object_ptr);
+			// In GroundWork Monitor, we suppress all deprecation warnings.
+			// We'll handle these issues via Monarch instead, so there's no
+			// point in worrying the end-user about them.
+			// logit(NSLOG_CONFIG_WARNING, TRUE, "%s", (char *)list->object_ptr);
 			free(list->object_ptr);
 			}
 		free_objectlist(&deprecated);
--- nagios-4.3.4/xdata/xodtemplate.c.orig	2017-08-24 14:43:48.000000000 -0700
+++ nagios-4.3.4/xdata/xodtemplate.c	2017-09-02 07:02:20.169176066 -0700
@@ -701,7 +701,10 @@
 					break;
 				}
 
-				logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: Extinfo objects are deprecated and will be removed in future versions\n");
+				// In GroundWork Monitor, we suppress all deprecation warnings.
+				// We'll handle these issues via Monarch instead, so there's no
+				// point in worrying the end-user about them.
+				// logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: Extinfo objects are deprecated and will be removed in future versions\n");
 			}
 
 			/* we're already in an object definition... */
@@ -2165,7 +2168,10 @@
 			else if(!strcmp(variable, "minimum_importance") ||
 					!strcmp(variable, "minimum_value")) {
 				if(!strcmp(variable, "minimum_value")) {
-					logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The minimum_value attribute is deprecated and will be removed in future versions. Please use minimum_importance instead.\n");
+					// In GroundWork Monitor, we suppress all deprecation warnings.
+					// We'll handle these issues via Monarch instead, so there's no
+					// point in worrying the end-user about them.
+					// logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The minimum_value attribute is deprecated and will be removed in future versions. Please use minimum_importance instead.\n");
 					}
 				temp_contact->minimum_value = strtoul(value, NULL, 10);
 				temp_contact->have_minimum_value = TRUE;
@@ -2408,21 +2414,32 @@
 				temp_host->have_initial_state = TRUE;
 				}
 			else if(!strcmp(variable, "check_interval") || !strcmp(variable, "normal_check_interval")) {
-				if(!strcmp(variable, "normal_check_interval"))
-					logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The normal_check_interval attribute is deprecated and will be removed in future versions. Please use check_interval instead.\n");
+				if(!strcmp(variable, "normal_check_interval")) {
+					// In GroundWork Monitor, we suppress all deprecation warnings.
+					// We'll handle these issues via Monarch instead, so there's no
+					// point in worrying the end-user about them.
+					// logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The normal_check_interval attribute is deprecated and will be removed in future versions. Please use check_interval instead.\n");
+					}
 				temp_host->check_interval = strtod(value, NULL);
 				temp_host->have_check_interval = TRUE;
 				}
 			else if(!strcmp(variable, "retry_interval") || !strcmp(variable, "retry_check_interval")) {
-				if(!strcmp(variable, "retry_check_interval"))
-					logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The retry_check_interval attribute is deprecated and will be removed in future versions. Please use retry_interval instead.\n");
+				if(!strcmp(variable, "retry_check_interval")) {
+					// In GroundWork Monitor, we suppress all deprecation warnings.
+					// We'll handle these issues via Monarch instead, so there's no
+					// point in worrying the end-user about them.
+					// logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The retry_check_interval attribute is deprecated and will be removed in future versions. Please use retry_interval instead.\n");
+					}
 				temp_host->retry_interval = strtod(value, NULL);
 				temp_host->have_retry_interval = TRUE;
 				}
 			else if(!strcmp(variable, "importance") ||
 					!strcmp(variable, "hourly_value")) {
 				if(!strcmp(variable, "hourly_value")) {
-					logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The hourly_value attribute is deprecated and will be removed in future versions. Please use importance instead.\n");
+					// In GroundWork Monitor, we suppress all deprecation warnings.
+					// We'll handle these issues via Monarch instead, so there's no
+					// point in worrying the end-user about them.
+					// logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The hourly_value attribute is deprecated and will be removed in future versions. Please use importance instead.\n");
 					}
 				temp_host->hourly_value = (unsigned int)strtoul(value, NULL, 10);
 				temp_host->have_hourly_value = 1;
@@ -2857,7 +2874,10 @@
 			else if(!strcmp(variable, "importance") ||
 					!strcmp(variable, "hourly_value")) {
 				if(!strcmp(variable, "hourly_value")) {
-					logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The hourly_value attribute is deprecated and will be removed in future versions. Please use importance instead.\n");
+					// In GroundWork Monitor, we suppress all deprecation warnings.
+					// We'll handle these issues via Monarch instead, so there's no
+					// point in worrying the end-user about them.
+					// logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The hourly_value attribute is deprecated and will be removed in future versions. Please use importance instead.\n");
 					}
 				temp_service->hourly_value = (unsigned int)strtoul(value, NULL, 10);
 				temp_service->have_hourly_value = 1;
@@ -2867,14 +2887,22 @@
 				temp_service->have_max_check_attempts = TRUE;
 				}
 			else if(!strcmp(variable, "check_interval") || !strcmp(variable, "normal_check_interval")) {
-				if(!strcmp(variable, "normal_check_interval"))
-					logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The normal_check_interval attribute is deprecated and will be removed in future versions. Please use check_interval instead.\n");
+				if(!strcmp(variable, "normal_check_interval")) {
+					// In GroundWork Monitor, we suppress all deprecation warnings.
+					// We'll handle these issues via Monarch instead, so there's no
+					// point in worrying the end-user about them.
+					// logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The normal_check_interval attribute is deprecated and will be removed in future versions. Please use check_interval instead.\n");
+					}
 				temp_service->check_interval = strtod(value, NULL);
 				temp_service->have_check_interval = TRUE;
 				}
 			else if(!strcmp(variable, "retry_interval") || !strcmp(variable, "retry_check_interval")) {
-				if(!strcmp(variable, "retry_check_interval"))
-					logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The retry_check_interval attribute is deprecated and will be removed in future versions. Please use retry_interval instead.\n");
+				if(!strcmp(variable, "retry_check_interval")) {
+					// In GroundWork Monitor, we suppress all deprecation warnings.
+					// We'll handle these issues via Monarch instead, so there's no
+					// point in worrying the end-user about them.
+					// logit(NSLOG_CONFIG_WARNING, TRUE, "WARNING: The retry_check_interval attribute is deprecated and will be removed in future versions. Please use retry_interval instead.\n");
+					}
 				temp_service->retry_interval = strtod(value, NULL);
 				temp_service->have_retry_interval = TRUE;
 				}
