--- nagios-4.3.4/common/macros.c.orig	2017-08-24 14:43:48.000000000 -0700
+++ nagios-4.3.4/common/macros.c	2017-09-03 10:25:01.344270254 -0700
@@ -1817,7 +1817,7 @@
 			break;
 		case MACRO_HOSTINFOURL:
 			buf1 = get_url_encoded_string(temp_host->name);
-			asprintf(output, "%s/cgi-bin/extinfo.cgi?type=1&host=%s",
+			asprintf(output, "%s/extinfo.cgi?type=1&host=%s",
 					website_url ? website_url : "website_url not set", buf1);
 			my_free(buf1);
 			break;
@@ -2157,7 +2157,7 @@
 
 			buf1 = get_url_encoded_string(temp_service->host_name);
 			buf2 = get_url_encoded_string(temp_service->description);
-			asprintf(output, "%s/cgi-bin/extinfo.cgi?type=2&host=%s&service=%s",
+			asprintf(output, "%s/extinfo.cgi?type=2&host=%s&service=%s",
 					website_url ? website_url : "website_url not set",
 					buf1, buf2);
 			my_free(buf1);
