--- nagios-3.2.3/base/checks.c.orig	2010-09-21 08:05:31.000000000 -0700
+++ nagios-3.2.3/base/checks.c	2011-01-11 11:18:40.000000000 -0800
@@ -1399,6 +1399,8 @@
 		/* 05/29/2007 NOTE: The host might be in a SOFT problem state due to host check retries/caching.  Not sure if we should take that into account and do something different or not... */
 		if(route_result!=HOST_UP){
 
+			log_debug_info(DEBUGL_CHECKS,2,"Host is not UP, so we mark state changes if appropriate\n");
+
 			/* "fake" a hard state change for the service - well, its not really fake, but it didn't get caught earlier... */
 			if(temp_service->last_hard_state!=temp_service->current_state)
 				hard_state_change=TRUE;
@@ -1406,8 +1408,11 @@
 			/* update last state change times */
 			if(state_change==TRUE || hard_state_change==TRUE)
 				temp_service->last_state_change=temp_service->last_check;
-			if(hard_state_change==TRUE)
+			if(hard_state_change==TRUE) {
 				temp_service->last_hard_state_change=temp_service->last_check;
+				temp_service->state_type=HARD_STATE;
+				temp_service->last_hard_state=temp_service->current_state;
+				}
 
 			/* put service into a hard state without attempting check retries and don't send out notifications about it */
 			temp_service->host_problem_at_last_check=TRUE;
